{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ACCESS BIOINFORMATICS DATABASES WITH BIO-PYTHON"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. [NCBI](#1.-NCBI)<br>\n",
    "    1.1. [Nucleotide BLAST](#1.1.-Nucleotide-BLAST)<br>\n",
    "    1.2. [Protein BLAST](#1.2.-Protein-BLAST)\n",
    "    \n",
    "2. [ENTREZ](#2.-ENTREZ)<br>\n",
    "    2.1. [PUBMED](#2.1.-PUBMED)<br>\n",
    "    2.2. [Nucleotide](#2.2.-Nucleotide)\n",
    "    \n",
    "3. [PDB](#3.-PDB)\n",
    "\n",
    "4. [EXPASY](#4.-EXPASY)<br>\n",
    "    4.1. [PROSITE](#4.1.-PROSITE)<br>\n",
    "    4.2. [ScanProsite](#4.2.-ScanProsite)\n",
    "    \n",
    "5. [KEGG](#5.-KEGG)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1. NCBI"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import Modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Blast import NCBIWWW\n",
    "from Bio import SeqIO, SearchIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on module Bio.Blast.NCBIWWW in Bio.Blast:\n",
      "\n",
      "NAME\n",
      "    Bio.Blast.NCBIWWW - Code to invoke the NCBI BLAST server over the internet.\n",
      "\n",
      "DESCRIPTION\n",
      "    This module provides code to work with the WWW version of BLAST\n",
      "    provided by the NCBI. https://blast.ncbi.nlm.nih.gov/\n",
      "\n",
      "FUNCTIONS\n",
      "    qblast(program, database, sequence, url_base='https://blast.ncbi.nlm.nih.gov/Blast.cgi', auto_format=None, composition_based_statistics=None, db_genetic_code=None, endpoints=None, entrez_query='(none)', expect=10.0, filter=None, gapcosts=None, genetic_code=None, hitlist_size=50, i_thresh=None, layout=None, lcase_mask=None, matrix_name=None, nucl_penalty=None, nucl_reward=None, other_advanced=None, perc_ident=None, phi_pattern=None, query_file=None, query_believe_defline=None, query_from=None, query_to=None, searchsp_eff=None, service=None, threshold=None, ungapped_alignment=None, word_size=None, short_query=None, alignments=500, alignment_view=None, descriptions=500, entrez_links_new_window=None, expect_low=None, expect_high=None, format_entrez_query=None, format_object=None, format_type='XML', ncbi_gi=None, results_file=None, show_overview=None, megablast=None, template_type=None, template_length=None)\n",
      "        BLAST search using NCBI's QBLAST server or a cloud service provider.\n",
      "        \n",
      "        Supports all parameters of the old qblast API for Put and Get.\n",
      "        \n",
      "        Please note that NCBI uses the new Common URL API for BLAST searches\n",
      "        on the internet (http://ncbi.github.io/blast-cloud/dev/api.html). Thus,\n",
      "        some of the parameters used by this function are not (or are no longer)\n",
      "        officially supported by NCBI. Although they are still functioning, this\n",
      "        may change in the future.\n",
      "        \n",
      "        The Common URL API (http://ncbi.github.io/blast-cloud/dev/api.html) allows\n",
      "        doing BLAST searches on cloud servers. To use this feature, please set\n",
      "        ``url_base='http://host.my.cloud.service.provider.com/cgi-bin/blast.cgi'``\n",
      "        and ``format_object='Alignment'``. For more details, please see\n",
      "        https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastDocs&DOC_TYPE=CloudBlast\n",
      "        \n",
      "        Some useful parameters:\n",
      "        \n",
      "         - program        blastn, blastp, blastx, tblastn, or tblastx (lower case)\n",
      "         - database       Which database to search against (e.g. \"nr\").\n",
      "         - sequence       The sequence to search.\n",
      "         - ncbi_gi        TRUE/FALSE whether to give 'gi' identifier.\n",
      "         - descriptions   Number of descriptions to show.  Def 500.\n",
      "         - alignments     Number of alignments to show.  Def 500.\n",
      "         - expect         An expect value cutoff.  Def 10.0.\n",
      "         - matrix_name    Specify an alt. matrix (PAM30, PAM70, BLOSUM80, BLOSUM45).\n",
      "         - filter         \"none\" turns off filtering.  Default no filtering\n",
      "         - format_type    \"HTML\", \"Text\", \"ASN.1\", or \"XML\".  Def. \"XML\".\n",
      "         - entrez_query   Entrez query to limit Blast search\n",
      "         - hitlist_size   Number of hits to return. Default 50\n",
      "         - megablast      TRUE/FALSE whether to use MEga BLAST algorithm (blastn only)\n",
      "         - short_query    TRUE/FALSE whether to adjust the search parameters for a\n",
      "                          short query sequence. Note that this will override\n",
      "                          manually set parameters like word size and e value. Turns\n",
      "                          off when sequence length is > 30 residues. Default: None.\n",
      "         - service        plain, psi, phi, rpsblast, megablast (lower case)\n",
      "        \n",
      "        This function does no checking of the validity of the parameters\n",
      "        and passes the values to the server as is.  More help is available at:\n",
      "        https://ncbi.github.io/blast-cloud/dev/api.html\n",
      "\n",
      "DATA\n",
      "    NCBI_BLAST_URL = 'https://blast.ncbi.nlm.nih.gov/Blast.cgi'\n",
      "\n",
      "FILE\n",
      "    /home/rhyme/.local/lib/python3.6/site-packages/Bio/Blast/NCBIWWW.py\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(NCBIWWW)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.1. Nucleotide BLAST"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dir  notebook.ipynb  nuc_seq.fasta  obsolete  prot_seq.fasta\r\n"
     ]
    }
   ],
   "source": [
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "774"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nuc_record=SeqIO.read(\"nuc_seq.fasta\",format=\"fasta\")\n",
    "len(nuc_record)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#nuc_record.description # description de sequance \n",
    "#nuc_rec_ord.seq  # seq de stering de nucloitde A C G T\n",
    "result_handle =NCBIWWW.qblast(\"blastn\",\"nt\",nuc_record.seq) # output in XML  format\n",
    "blast_result =SearchIO.read(result_handle,\"blast-xml\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Program: blastn (2.15.0+)\n",
      "  Query: No (774)\n",
      "         definition line\n",
      " Target: nt\n",
      "   Hits: ----  -----  ----------------------------------------------------------\n",
      "            #  # HSP  ID + description\n",
      "         ----  -----  ----------------------------------------------------------\n",
      "            0      1  gi|2716167577|gb|PP645709.1|  Severe acute respiratory ...\n",
      "            1      1  gi|2716164760|gb|PP645492.1|  Severe acute respiratory ...\n"
     ]
    }
   ],
   "source": [
    "print(blast_result[0:2])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sequence ID: gi|2716167577|gb|PP645709.1|\n",
      "Sequence Description: Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/human/USA/RI-RISHL-023586/2020 ORF1ab polyprotein (ORF1ab), ORF1a polyprotein (ORF1ab), surface glycoprotein (S), ORF3a protein (ORF3a), envelope protein (E), membrane glycoprotein (M), ORF6 protein (ORF6), ORF7a protein (ORF7a), ORF7b (ORF7b), ORF8 protein (ORF8), nucleocapsid phosphoprotein (N), and ORF10 protein (ORF10) genes, complete cds\n",
      "E-value: 0.0\n"
     ]
    }
   ],
   "source": [
    "Seq = blast_result[0]\n",
    "print(f\"Sequence ID: {Seq.id}\")\n",
    "print(f\"Sequence Description: {Seq.description}\")\n",
    "\n",
    "details=Seq[0]\n",
    "print(f\"E-value: {details.evalue}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "alignment:\n",
      "Alignment with 2 rows and 774 columns\n",
      "ATCGCTCCAGGGCAAACTGGAAAGATTGCTGATTATAATTATAA...GGT No\n",
      "ATCGCTCCAGGGCAAACTGGAAAGATTGCTGATTATAATTATAA...GGT gi|2716167577|gb|PP645709.1|\n"
     ]
    }
   ],
   "source": [
    "print(f\"alignment:\\n{details.aln}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.2. Protein BLAST"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "258"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_record = SeqIO.read(\"prot_seq.fasta\", format=\"fasta\")\n",
    "len(prot_record)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "result_handle = NCBIWWW.qblast(\"blastp\", \"pdb\", prot_record.seq)\n",
    "blast_result = SearchIO.read(result_handle, \"blast-xml\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Program: blastp (2.15.0+)\n",
      "  Query: unnamed (258)\n",
      "         protein product\n",
      " Target: pdb\n",
      "   Hits: ----  -----  ----------------------------------------------------------\n",
      "            #  # HSP  ID + description\n",
      "         ----  -----  ----------------------------------------------------------\n",
      "            0      1  pdb|8ELJ|A  Chain A, Spike glycoprotein [Severe acute r...\n",
      "            1      1  pdb|7CAB|A  Chain A, Spike glycoprotein [Severe acute r...\n"
     ]
    }
   ],
   "source": [
    "print(blast_result[0:2])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sequence ID: pdb|8ELJ|A\n",
      "Sequence Description: Chain A, Spike glycoprotein [Severe acute respiratory syndrome coronavirus 2]\n",
      "E-value: 0.0\n"
     ]
    }
   ],
   "source": [
    "Seq = blast_result [0]\n",
    "print(f\"Sequence ID: {Seq.id}\")\n",
    "print(f\"Sequence Description: {Seq.description}\")\n",
    "\n",
    "details = Seq[0]\n",
    "print(f\"E-value: {details.evalue}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "alignment:\n",
      " Alignment with 2 rows and 258 columns\n",
      "IAPGQTGKIADYNYKLPDDFTGCVIAWNSNNLDSKVGGNYNYLY...PIG unnamed\n",
      "IAPGQTGKIADYNYKLPDDFTGCVIAWNSNNLDSKVGGNYNYLY...PIG pdb|8ELJ|A\n"
     ]
    }
   ],
   "source": [
    "print(f\"alignment:\\n {details.aln}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "------------------------------------------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 2. ENTREZ"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import Modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on package Bio.Entrez in Bio:\n",
      "\n",
      "NAME\n",
      "    Bio.Entrez - Provides code to access NCBI over the WWW.\n",
      "\n",
      "DESCRIPTION\n",
      "    The main Entrez web page is available at:\n",
      "    http://www.ncbi.nlm.nih.gov/Entrez/\n",
      "    \n",
      "    Entrez Programming Utilities web page is available at:\n",
      "    http://www.ncbi.nlm.nih.gov/books/NBK25501/\n",
      "    \n",
      "    This module provides a number of functions like ``efetch`` (short for\n",
      "    Entrez Fetch) which will return the data as a handle object. This is\n",
      "    a standard interface used in Python for reading data from a file, or\n",
      "    in this case a remote network connection, and provides methods like\n",
      "    ``.read()`` or offers iteration over the contents line by line. See\n",
      "    also \"What the heck is a handle?\" in the Biopython Tutorial and\n",
      "    Cookbook: http://biopython.org/DIST/docs/tutorial/Tutorial.html\n",
      "    http://biopython.org/DIST/docs/tutorial/Tutorial.pdf\n",
      "    The handle returned by these functions can be either in text mode or\n",
      "    in binary mode, depending on the data requested and the results\n",
      "    returned by NCBI Entrez. Typically, XML data will be in binary mode\n",
      "    while other data will be in text mode, as required by the downstream\n",
      "    parser to parse the data.\n",
      "    \n",
      "    Unlike a handle to a file on disk from the ``open(filename)`` function,\n",
      "    which has a ``.name`` attribute giving the filename, the handles from\n",
      "    ``Bio.Entrez`` all have a ``.url`` attribute instead giving the URL\n",
      "    used to connect to the NCBI Entrez API.\n",
      "    \n",
      "    All the functions that send requests to the NCBI Entrez API will\n",
      "    automatically respect the NCBI rate limit (of 3 requests per second\n",
      "    without an API key, or 10 requests per second with an API key) and\n",
      "    will automatically retry when encountering transient failures\n",
      "    (i.e. connection failures or HTTP 5XX codes). By default, Biopython\n",
      "    does a maximum of three tries before giving up, and sleeps for 15\n",
      "    seconds between tries. You can tweak these parameters by setting\n",
      "    ``Bio.Entrez.max_tries`` and ``Bio.Entrez.sleep_between_tries``.\n",
      "    \n",
      "    The Entrez module also provides an XML parser which takes a handle\n",
      "    as input.\n",
      "    \n",
      "    Variables:\n",
      "    \n",
      "        - email        Set the Entrez email parameter (default is not set).\n",
      "        - tool         Set the Entrez tool parameter (default is ``biopython``).\n",
      "        - api_key      Personal API key from NCBI. If not set, only 3 queries per\n",
      "          second are allowed. 10 queries per seconds otherwise with a\n",
      "          valid API key.\n",
      "        - max_tries    Configures how many times failed requests will be\n",
      "          automatically retried on error (default is 3).\n",
      "        - sleep_between_tries   The delay, in seconds, before retrying a request on\n",
      "          error (default is 15).\n",
      "    \n",
      "    Functions:\n",
      "    \n",
      "        - efetch       Retrieves records in the requested format from a list of one or\n",
      "          more primary IDs or from the user's environment\n",
      "        - epost        Posts a file containing a list of primary IDs for future use in\n",
      "          the user's environment to use with subsequent search strategies\n",
      "        - esearch      Searches and retrieves primary IDs (for use in EFetch, ELink,\n",
      "          and ESummary) and term translations and optionally retains\n",
      "          results for future use in the user's environment.\n",
      "        - elink        Checks for the existence of an external or Related Articles link\n",
      "          from a list of one or more primary IDs.  Retrieves primary IDs\n",
      "          and relevancy scores for links to Entrez databases or Related\n",
      "          Articles;  creates a hyperlink to the primary LinkOut provider\n",
      "          for a specific ID and database, or lists LinkOut URLs\n",
      "          and Attributes for multiple IDs.\n",
      "        - einfo        Provides field index term counts, last update, and available\n",
      "          links for each database.\n",
      "        - esummary     Retrieves document summaries from a list of primary IDs or from\n",
      "          the user's environment.\n",
      "        - egquery      Provides Entrez database counts in XML for a single search\n",
      "          using Global Query.\n",
      "        - espell       Retrieves spelling suggestions.\n",
      "        - ecitmatch    Retrieves PubMed IDs (PMIDs) that correspond to a set of\n",
      "          input citation strings.\n",
      "    \n",
      "        - read         Parses the XML results returned by any of the above functions.\n",
      "          Alternatively, the XML data can be read from a file opened in binary mode.\n",
      "          Typical usage is:\n",
      "    \n",
      "              >>> from Bio import Entrez\n",
      "              >>> Entrez.email = \"Your.Name.Here@example.org\"\n",
      "              >>> handle = Entrez.einfo() # or esearch, efetch, ...\n",
      "              >>> record = Entrez.read(handle)\n",
      "              >>> handle.close()\n",
      "    \n",
      "           where record is now a Python dictionary or list.\n",
      "    \n",
      "        - parse        Parses the XML results returned by those of the above functions\n",
      "          which can return multiple records - such as efetch, esummary\n",
      "          and elink. Typical usage is:\n",
      "    \n",
      "              >>> handle = Entrez.esummary(db=\"pubmed\", id=\"19304878,14630660\", retmode=\"xml\")\n",
      "              >>> records = Entrez.parse(handle)\n",
      "              >>> for record in records:\n",
      "              ...     # each record is a Python dictionary or list.\n",
      "              ...     print(record['Title'])\n",
      "              Biopython: freely available Python tools for computational molecular biology and bioinformatics.\n",
      "              PDB file parser and structure class implemented in Python.\n",
      "              >>> handle.close()\n",
      "    \n",
      "          This function is appropriate only if the XML file contains\n",
      "          multiple records, and is particular useful for large files.\n",
      "    \n",
      "        - _open        Internally used function.\n",
      "\n",
      "PACKAGE CONTENTS\n",
      "    Parser\n",
      "\n",
      "FUNCTIONS\n",
      "    ecitmatch(**keywds)\n",
      "        Retrieve PMIDs for input citation strings, returned as a handle.\n",
      "        \n",
      "        ECitMatch retrieves PubMed IDs (PMIDs) that correspond to a set of input\n",
      "        citation strings.\n",
      "        \n",
      "        See the online documentation for an explanation of the parameters:\n",
      "        http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ECitMatch\n",
      "        \n",
      "        Return a handle to the results, by default in plain text\n",
      "        \n",
      "        Raises an IOError exception if there's a network error.\n",
      "        \n",
      "        Short example:\n",
      "        \n",
      "        >>> from Bio import Entrez\n",
      "        >>> Entrez.email = \"Your.Name.Here@example.org\"\n",
      "        >>> citation_1 = {\"journal_title\": \"proc natl acad sci u s a\",\n",
      "        ...               \"year\": \"1991\", \"volume\": \"88\", \"first_page\": \"3248\",\n",
      "        ...               \"author_name\": \"mann bj\", \"key\": \"citation_1\"}\n",
      "        >>> handle = Entrez.ecitmatch(db=\"pubmed\", bdata=[citation_1])\n",
      "        >>> print(handle.read().strip().split(\"|\"))\n",
      "        ['proc natl acad sci u s a', '1991', '88', '3248', 'mann bj', 'citation_1', '2014248']\n",
      "        >>> handle.close()\n",
      "    \n",
      "    efetch(db, **keywords)\n",
      "        Fetch Entrez results which are returned as a handle.\n",
      "        \n",
      "        EFetch retrieves records in the requested format from a list or set of one or\n",
      "        more UIs or from user's environment.\n",
      "        \n",
      "        See the online documentation for an explanation of the parameters:\n",
      "        http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EFetch\n",
      "        \n",
      "        Return a handle to the results.\n",
      "        \n",
      "        Raises an IOError exception if there's a network error.\n",
      "        \n",
      "        Short example:\n",
      "        \n",
      "        >>> from Bio import Entrez\n",
      "        >>> Entrez.email = \"Your.Name.Here@example.org\"\n",
      "        >>> handle = Entrez.efetch(db=\"nucleotide\", id=\"AY851612\", rettype=\"gb\", retmode=\"text\")\n",
      "        >>> print(handle.readline().strip())\n",
      "        LOCUS       AY851612                 892 bp    DNA     linear   PLN 10-APR-2007\n",
      "        >>> handle.close()\n",
      "        \n",
      "        This will automatically use an HTTP POST rather than HTTP GET if there\n",
      "        are over 200 identifiers as recommended by the NCBI.\n",
      "        \n",
      "        **Warning:** The NCBI changed the default retmode in Feb 2012, so many\n",
      "        databases which previously returned text output now give XML.\n",
      "    \n",
      "    egquery(**keywds)\n",
      "        Provide Entrez database counts for a global search.\n",
      "        \n",
      "        EGQuery provides Entrez database counts in XML for a single search\n",
      "        using Global Query.\n",
      "        \n",
      "        See the online documentation for an explanation of the parameters:\n",
      "        http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EGQuery\n",
      "        \n",
      "        Return a handle to the results in XML format.\n",
      "        \n",
      "        Raises an IOError exception if there's a network error.\n",
      "        \n",
      "        This quick example based on a longer version from the Biopython\n",
      "        Tutorial just checks there are over 60 matches for 'Biopython'\n",
      "        in PubMedCentral:\n",
      "        \n",
      "        >>> from Bio import Entrez\n",
      "        >>> Entrez.email = \"Your.Name.Here@example.org\"\n",
      "        >>> handle = Entrez.egquery(term=\"biopython\")\n",
      "        >>> record = Entrez.read(handle)\n",
      "        >>> handle.close()\n",
      "        >>> for row in record[\"eGQueryResult\"]:\n",
      "        ...     if \"pmc\" in row[\"DbName\"]:\n",
      "        ...         print(int(row[\"Count\"]) > 60)\n",
      "        True\n",
      "    \n",
      "    einfo(**keywds)\n",
      "        Return a summary of the Entrez databases as a results handle.\n",
      "        \n",
      "        EInfo provides field names, index term counts, last update, and\n",
      "        available links for each Entrez database.\n",
      "        \n",
      "        See the online documentation for an explanation of the parameters:\n",
      "        http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EInfo\n",
      "        \n",
      "        Return a handle to the results, by default in XML format.\n",
      "        \n",
      "        Raises an IOError exception if there's a network error.\n",
      "        \n",
      "        Short example:\n",
      "        \n",
      "        >>> from Bio import Entrez\n",
      "        >>> Entrez.email = \"Your.Name.Here@example.org\"\n",
      "        >>> record = Entrez.read(Entrez.einfo())\n",
      "        >>> 'pubmed' in record['DbList']\n",
      "        True\n",
      "    \n",
      "    elink(**keywds)\n",
      "        Check for linked external articles and return a handle.\n",
      "        \n",
      "        ELink checks for the existence of an external or Related Articles link\n",
      "        from a list of one or more primary IDs;  retrieves IDs and relevancy\n",
      "        scores for links to Entrez databases or Related Articles; creates a\n",
      "        hyperlink to the primary LinkOut provider for a specific ID and\n",
      "        database, or lists LinkOut URLs and attributes for multiple IDs.\n",
      "        \n",
      "        See the online documentation for an explanation of the parameters:\n",
      "        http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ELink\n",
      "        \n",
      "        Return a handle to the results, by default in XML format.\n",
      "        \n",
      "        Raises an IOError exception if there's a network error.\n",
      "        \n",
      "        This example finds articles related to the Biopython application\n",
      "        note's entry in the PubMed database:\n",
      "        \n",
      "        >>> from Bio import Entrez\n",
      "        >>> Entrez.email = \"Your.Name.Here@example.org\"\n",
      "        >>> pmid = \"19304878\"\n",
      "        >>> handle = Entrez.elink(dbfrom=\"pubmed\", id=pmid, linkname=\"pubmed_pubmed\")\n",
      "        >>> record = Entrez.read(handle)\n",
      "        >>> handle.close()\n",
      "        >>> print(record[0][\"LinkSetDb\"][0][\"LinkName\"])\n",
      "        pubmed_pubmed\n",
      "        >>> linked = [link[\"Id\"] for link in record[0][\"LinkSetDb\"][0][\"Link\"]]\n",
      "        >>> \"17121776\" in linked\n",
      "        True\n",
      "        \n",
      "        This is explained in much more detail in the Biopython Tutorial.\n",
      "    \n",
      "    epost(db, **keywds)\n",
      "        Post a file of identifiers for future use.\n",
      "        \n",
      "        Posts a file containing a list of UIs for future use in the user's\n",
      "        environment to use with subsequent search strategies.\n",
      "        \n",
      "        See the online documentation for an explanation of the parameters:\n",
      "        http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EPost\n",
      "        \n",
      "        Return a handle to the results.\n",
      "        \n",
      "        Raises an IOError exception if there's a network error.\n",
      "    \n",
      "    esearch(db, term, **keywds)\n",
      "        Run an Entrez search and return a handle to the results.\n",
      "        \n",
      "        ESearch searches and retrieves primary IDs (for use in EFetch, ELink\n",
      "        and ESummary) and term translations, and optionally retains results\n",
      "        for future use in the user's environment.\n",
      "        \n",
      "        See the online documentation for an explanation of the parameters:\n",
      "        http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ESearch\n",
      "        \n",
      "        Return a handle to the results which are always in XML format.\n",
      "        \n",
      "        Raises an IOError exception if there's a network error.\n",
      "        \n",
      "        Short example:\n",
      "        \n",
      "        >>> from Bio import Entrez\n",
      "        >>> Entrez.email = \"Your.Name.Here@example.org\"\n",
      "        >>> handle = Entrez.esearch(db=\"nucleotide\", retmax=10, term=\"opuntia[ORGN] accD\", idtype=\"acc\")\n",
      "        >>> record = Entrez.read(handle)\n",
      "        >>> handle.close()\n",
      "        >>> int(record[\"Count\"]) >= 2\n",
      "        True\n",
      "        >>> \"EF590893.1\" in record[\"IdList\"]\n",
      "        True\n",
      "        >>> \"EF590892.1\" in record[\"IdList\"]\n",
      "        True\n",
      "    \n",
      "    espell(**keywds)\n",
      "        Retrieve spelling suggestions as a results handle.\n",
      "        \n",
      "        ESpell retrieves spelling suggestions, if available.\n",
      "        \n",
      "        See the online documentation for an explanation of the parameters:\n",
      "        http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ESpell\n",
      "        \n",
      "        Return a handle to the results, by default in XML format.\n",
      "        \n",
      "        Raises an IOError exception if there's a network error.\n",
      "        \n",
      "        Short example:\n",
      "        \n",
      "        >>> from Bio import Entrez\n",
      "        >>> Entrez.email = \"Your.Name.Here@example.org\"\n",
      "        >>> record = Entrez.read(Entrez.espell(term=\"biopythooon\"))\n",
      "        >>> print(record[\"Query\"])\n",
      "        biopythooon\n",
      "        >>> print(record[\"CorrectedQuery\"])\n",
      "        biopython\n",
      "    \n",
      "    esummary(**keywds)\n",
      "        Retrieve document summaries as a results handle.\n",
      "        \n",
      "        ESummary retrieves document summaries from a list of primary IDs or\n",
      "        from the user's environment.\n",
      "        \n",
      "        See the online documentation for an explanation of the parameters:\n",
      "        http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ESummary\n",
      "        \n",
      "        Return a handle to the results, by default in XML format.\n",
      "        \n",
      "        Raises an IOError exception if there's a network error.\n",
      "        \n",
      "        This example discovers more about entry 19923 in the structure\n",
      "        database:\n",
      "        \n",
      "        >>> from Bio import Entrez\n",
      "        >>> Entrez.email = \"Your.Name.Here@example.org\"\n",
      "        >>> handle = Entrez.esummary(db=\"structure\", id=\"19923\")\n",
      "        >>> record = Entrez.read(handle)\n",
      "        >>> handle.close()\n",
      "        >>> print(record[0][\"Id\"])\n",
      "        19923\n",
      "        >>> print(record[0][\"PdbDescr\"])\n",
      "        Crystal Structure Of E. Coli Aconitase B\n",
      "    \n",
      "    parse(handle, validate=True, escape=False)\n",
      "        Parse an XML file from the NCBI Entrez Utilities into python objects.\n",
      "        \n",
      "        This function parses an XML file created by NCBI's Entrez Utilities,\n",
      "        returning a multilevel data structure of Python lists and dictionaries.\n",
      "        This function is suitable for XML files that (in Python) can be represented\n",
      "        as a list of individual records. Whereas 'read' reads the complete file\n",
      "        and returns a single Python list, 'parse' is a generator function that\n",
      "        returns the records one by one. This function is therefore particularly\n",
      "        useful for parsing large files.\n",
      "        \n",
      "        Most XML files returned by NCBI's Entrez Utilities can be parsed by\n",
      "        this function, provided its DTD is available. Biopython includes the\n",
      "        DTDs for most commonly used Entrez Utilities.\n",
      "        \n",
      "        The handle must be in binary mode. This allows the parser to detect the\n",
      "        encoding from the XML file, and to use it to convert all text in the XML\n",
      "        to the correct Unicode string. The functions in Bio.Entrez to access NCBI\n",
      "        Entrez will automatically return XML data in binary mode. For files,\n",
      "        please use mode \"rb\" when opening the file, as in\n",
      "        \n",
      "            >>> from Bio import Entrez\n",
      "            >>> handle = open(\"Entrez/pubmed1.xml\", \"rb\")  # opened in binary mode\n",
      "            >>> records = Entrez.parse(handle)\n",
      "            >>> for record in records:\n",
      "            ...     print(record['MedlineCitation']['Article']['Journal']['Title'])\n",
      "            ...\n",
      "            Social justice (San Francisco, Calif.)\n",
      "            Biochimica et biophysica acta\n",
      "            >>> handle.close()\n",
      "        \n",
      "        If validate is True (default), the parser will validate the XML file\n",
      "        against the DTD, and raise an error if the XML file contains tags that\n",
      "        are not represented in the DTD. If validate is False, the parser will\n",
      "        simply skip such tags.\n",
      "        \n",
      "        If escape is True, all characters that are not valid HTML are replaced\n",
      "        by HTML escape characters to guarantee that the returned strings are\n",
      "        valid HTML fragments. For example, a less-than sign (<) is replaced by\n",
      "        &lt;. If escape is False (default), the string is returned as is.\n",
      "        \n",
      "        Whereas the data structure seems to consist of generic Python lists,\n",
      "        dictionaries, strings, and so on, each of these is actually a class\n",
      "        derived from the base type. This allows us to store the attributes\n",
      "        (if any) of each element in a dictionary my_element.attributes, and\n",
      "        the tag name in my_element.tag.\n",
      "    \n",
      "    read(handle, validate=True, escape=False)\n",
      "        Parse an XML file from the NCBI Entrez Utilities into python objects.\n",
      "        \n",
      "        This function parses an XML file created by NCBI's Entrez Utilities,\n",
      "        returning a multilevel data structure of Python lists and dictionaries.\n",
      "        Most XML files returned by NCBI's Entrez Utilities can be parsed by\n",
      "        this function, provided its DTD is available. Biopython includes the\n",
      "        DTDs for most commonly used Entrez Utilities.\n",
      "        \n",
      "        The handle must be in binary mode. This allows the parser to detect the\n",
      "        encoding from the XML file, and to use it to convert all text in the XML\n",
      "        to the correct Unicode string. The functions in Bio.Entrez to access NCBI\n",
      "        Entrez will automatically return XML data in binary mode. For files,\n",
      "        please use mode \"rb\" when opening the file, as in\n",
      "        \n",
      "            >>> from Bio import Entrez\n",
      "            >>> handle = open(\"Entrez/esearch1.xml\", \"rb\")  # opened in binary mode\n",
      "            >>> record = Entrez.read(handle)\n",
      "            >>> print(record['QueryTranslation'])\n",
      "            biopython[All Fields]\n",
      "            >>> handle.close()\n",
      "        \n",
      "        If validate is True (default), the parser will validate the XML file\n",
      "        against the DTD, and raise an error if the XML file contains tags that\n",
      "        are not represented in the DTD. If validate is False, the parser will\n",
      "        simply skip such tags.\n",
      "        \n",
      "        If escape is True, all characters that are not valid HTML are replaced\n",
      "        by HTML escape characters to guarantee that the returned strings are\n",
      "        valid HTML fragments. For example, a less-than sign (<) is replaced by\n",
      "        &lt;. If escape is False (default), the string is returned as is.\n",
      "        \n",
      "        Whereas the data structure seems to consist of generic Python lists,\n",
      "        dictionaries, strings, and so on, each of these is actually a class\n",
      "        derived from the base type. This allows us to store the attributes\n",
      "        (if any) of each element in a dictionary my_element.attributes, and\n",
      "        the tag name in my_element.tag.\n",
      "\n",
      "DATA\n",
      "    api_key = None\n",
      "    email = None\n",
      "    max_tries = 3\n",
      "    sleep_between_tries = 15\n",
      "    tool = 'biopython'\n",
      "\n",
      "FILE\n",
      "    /home/rhyme/.local/lib/python3.6/site-packages/Bio/Entrez/__init__.py\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(Entrez)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "Entrez.email = \"datacyclopes@gmail.com\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['pubmed', 'protein', 'nuccore', 'ipg', 'nucleotide', 'structure', 'genome', 'annotinfo', 'assembly', 'bioproject', 'biosample', 'blastdbinfo', 'books', 'cdd', 'clinvar', 'gap', 'gapplus', 'grasp', 'dbvar', 'gene', 'gds', 'geoprofiles', 'medgen', 'mesh', 'nlmcatalog', 'omim', 'orgtrack', 'pmc', 'popset', 'proteinclusters', 'pcassay', 'protfam', 'pccompound', 'pcsubstance', 'seqannot', 'snp', 'sra', 'taxonomy', 'biocollections', 'gtr']"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "handle = Entrez.einfo()\n",
    "record = Entrez.read(handle)\n",
    "record[\"DbList\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.1. PUBMED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'PubMed bibliographic record'"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "handle = Entrez.einfo(db=\"pubmed\")\n",
    "record = Entrez.read(handle)\n",
    "record[\"DbInfo\"][\"Description\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'37208799'"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record[\"DbInfo\"][\"Count\"]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['38650605', '38365590', '38235175', '37810457', '37668712', '36818783', '36245797', '36094101', '35497637', '35496474', '35402671', '34735950', '34484417', '34434786', '34189012', '33994075', '33902722', '33809815', '33242467', '32044951']"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "handle = Entrez.esearch(db=\"pubmed\", term=\"biopython\")\n",
    "record = Entrez.read(handle)\n",
    "record[\"IdList\"]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Sulkowski A', 'Bouton C', 'Swanson C'] Syn-CpG-Spacer: A Panel web app for synonymous recoding of viral genomes with CpG dinucleotides. 2024 Apr 3 Journal of open source software\n",
      "['Bessa-Silva A'] Fasta2Structure: a user-friendly tool for converting multiple aligned FASTA files to STRUCTURE format. 2024 Feb 15 BMC bioinformatics\n"
     ]
    }
   ],
   "source": [
    "handle = Entrez.esummary(db=\"pubmed\", id='38650605, 38365590')\n",
    "records = Entrez.parse(handle)\n",
    "\n",
    "\n",
    "for record in records:\n",
    "    print(record['AuthorList'],record['Title'],record['PubDate'],record['FullJournalName'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "b'<?xml version=\"1.0\" ?>\\n<!DOCTYPE PubmedArticleSet PUBLIC \"-//NLM//DTD PubMedArticle, 1st January 2024//EN\" \"https://dtd.nlm.nih.gov/ncbi/pubmed/out/pubmed_240101.dtd\">\\n<PubmedArticleSet>\\n<PubmedArticle><MedlineCitation Status=\"MEDLINE\" Owner=\"NLM\"><PMID Version=\"1\">19811691</PMID><DateCompleted><Year>2010</Year><Month>02</Month><Day>12</Day></DateCompleted><DateRevised><Year>2021</Year><Month>10</Month><Day>20</Day></DateRevised><Article PubModel=\"Electronic\"><Journal><ISSN IssnType=\"Electronic\">1471-2105</ISSN><JournalIssue CitedMedium=\"Internet\"><Volume>10 Suppl 11</Volume><Issue>Suppl 11</Issue><PubDate><Year>2009</Year><Month>Oct</Month><Day>08</Day></PubDate></JournalIssue><Title>BMC bioinformatics</Title><ISOAbbreviation>BMC Bioinformatics</ISOAbbreviation></Journal><ArticleTitle>Exploratory visual analysis of conserved domains on multiple sequence alignments.</ArticleTitle><Pagination><StartPage>S7</StartPage><MedlinePgn>S7</MedlinePgn></Pagination><ELocationID EIdType=\"doi\" ValidYN=\"Y\">10.1186/1471-2105-10-S11-S7</ELocationID><Abstract><AbstractText Label=\"BACKGROUND\" NlmCategory=\"BACKGROUND\">Multiple alignment of protein sequences can provide insight into sequence conservation across many species and thus allow identification of those sections of the sequence most critical to protein function. This insight can be augmented by joint display of conserved domains along the sequences. By fusing this metadata visually, biologists can analyze sequence conservation and functional motifs simultaneously and efficiently.</AbstractText><AbstractText Label=\"RESULTS\" NlmCategory=\"RESULTS\">We present MSAVis, a new approach combining luminance and hue for simultaneous visualization of conserved motifs and sequence alignment. Input for the algorithm is a multiple sequence alignment in a standard format. The NCBI Conserved Domain Database (CDD) is used for finding conserved domains along the alignment. The visualization quickly identifies conserved domains, and allows both macro (sequence-long) and micro (small amino-acid neighborhood) views.</AbstractText><AbstractText Label=\"CONCLUSION\" NlmCategory=\"CONCLUSIONS\">MSAVis utilizes two visual cues, luminance and hue, to facilitate at-a-glance summary of the conservation of a user-provided protein alignment while enabling multiple comparisons among functional domains. These visual cues are preattentive and separable so that the relationship between conservation strength and domain membership can be understood. The MSAVis software, written in Python and using BioPython and OpenGL, can be found at http://agbase.msstate.edu/tools/MSAVis.html.</AbstractText></Abstract><AuthorList CompleteYN=\"Y\"><Author ValidYN=\"Y\"><LastName>Jankun-Kelly</LastName><ForeName>T J</ForeName><Initials>TJ</Initials><AffiliationInfo><Affiliation>Institute for Digital Biology and Department of Computer Science and Engineering, Bagley College of Engineering, Mississippi State University, Mississippi, USA. tjk@acm.org</Affiliation></AffiliationInfo></Author><Author ValidYN=\"Y\"><LastName>Lindeman</LastName><ForeName>Andrew D</ForeName><Initials>AD</Initials></Author><Author ValidYN=\"Y\"><LastName>Bridges</LastName><ForeName>Susan M</ForeName><Initials>SM</Initials></Author></AuthorList><Language>eng</Language><PublicationTypeList><PublicationType UI=\"D016428\">Journal Article</PublicationType><PublicationType UI=\"D013486\">Research Support, U.S. Gov\\'t, Non-P.H.S.</PublicationType></PublicationTypeList><ArticleDate DateType=\"Electronic\"><Year>2009</Year><Month>10</Month><Day>08</Day></ArticleDate></Article><MedlineJournalInfo><Country>England</Country><MedlineTA>BMC Bioinformatics</MedlineTA><NlmUniqueID>100965194</NlmUniqueID><ISSNLinking>1471-2105</ISSNLinking></MedlineJournalInfo><ChemicalList><Chemical><RegistryNumber>0</RegistryNumber><NameOfSubstance UI=\"D011506\">Proteins</NameOfSubstance></Chemical></ChemicalList><CitationSubset>IM</CitationSubset><MeshHeadingList><MeshHeading><DescriptorName UI=\"D000595\" MajorTopicYN=\"N\">Amino Acid Sequence</DescriptorName></MeshHeading><MeshHeading><DescriptorName UI=\"D019295\" MajorTopicYN=\"N\">Computational Biology</DescriptorName><QualifierName UI=\"Q000379\" MajorTopicYN=\"Y\">methods</QualifierName></MeshHeading><MeshHeading><DescriptorName UI=\"D030562\" MajorTopicYN=\"N\">Databases, Protein</DescriptorName></MeshHeading><MeshHeading><DescriptorName UI=\"D008969\" MajorTopicYN=\"N\">Molecular Sequence Data</DescriptorName></MeshHeading><MeshHeading><DescriptorName UI=\"D017434\" MajorTopicYN=\"Y\">Protein Structure, Tertiary</DescriptorName></MeshHeading><MeshHeading><DescriptorName UI=\"D011506\" MajorTopicYN=\"N\">Proteins</DescriptorName><QualifierName UI=\"Q000737\" MajorTopicYN=\"N\">chemistry</QualifierName></MeshHeading><MeshHeading><DescriptorName UI=\"D016415\" MajorTopicYN=\"Y\">Sequence Alignment</DescriptorName></MeshHeading><MeshHeading><DescriptorName UI=\"D020539\" MajorTopicYN=\"N\">Sequence Analysis, Protein</DescriptorName><QualifierName UI=\"Q000379\" MajorTopicYN=\"Y\">methods</QualifierName></MeshHeading></MeshHeadingList></MedlineCitation><PubmedData><History><PubMedPubDate PubStatus=\"entrez\"><Year>2009</Year><Month>10</Month><Day>9</Day><Hour>6</Hour><Minute>0</Minute></PubMedPubDate><PubMedPubDate PubStatus=\"pubmed\"><Year>2009</Year><Month>10</Month><Day>14</Day><Hour>6</Hour><Minute>0</Minute></PubMedPubDate><PubMedPubDate PubStatus=\"medline\"><Year>2010</Year><Month>2</Month><Day>13</Day><Hour>6</Hour><Minute>0</Minute></PubMedPubDate><PubMedPubDate PubStatus=\"pmc-release\"><Year>2009</Year><Month>10</Month><Day>8</Day></PubMedPubDate></History><PublicationStatus>epublish</PublicationStatus><ArticleIdList><ArticleId IdType=\"pubmed\">19811691</ArticleId><ArticleId IdType=\"pmc\">PMC3226196</ArticleId><ArticleId IdType=\"doi\">10.1186/1471-2105-10-S11-S7</ArticleId><ArticleId IdType=\"pii\">1471-2105-10-S11-S7</ArticleId></ArticleIdList><ReferenceList><Reference><Citation>Larkin M, Blackshields G, Brown N, Chenna R, McGettigan P, McWilliam H, Valentin F, Wallace I, Wilm A, Lopez R, Thompson J, Gibson T, Higgins D. ClustalW and ClustalX version 2. Bioinformatics. 2007;23(21):2947&#x2013;2948. doi: 10.1093/bioinformatics/btm404.</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1093/bioinformatics/btm404</ArticleId><ArticleId IdType=\"pubmed\">17846036</ArticleId></ArticleIdList></Reference><Reference><Citation>A Conserved Domain Database and Search Service. http://www.ncbi.nlm.nih.gov/Structure/cdd/cdd.shtml</Citation></Reference><Reference><Citation>Marchler-Bauer A, Anderson J, Cherukuri P, DeWeese-Scott C, Geer L, Gwadz M, He S, Hurwitz D, Jackson J, Ke Z, Lanczycki C, Liebert C, Liu C, Lu F, Marchler G, Mullokandov M, Shoemaker B, Simonyan V, Song J, Thiessen P, Yamashita R, Yin J, Zhang D, Bryant S. CDD: A Conserved Domain Database for protein classification. Nucleic Acids Research. 2005. pp. D192&#x2013;D196.</Citation><ArticleIdList><ArticleId IdType=\"pmc\">PMC540023</ArticleId><ArticleId IdType=\"pubmed\">15608175</ArticleId></ArticleIdList></Reference><Reference><Citation>Lungu M, Xu K.  In: Human-Centered Visualization Environments, Volume 4417 of Lecture Notes in Computer Science. Kerren A, Ebert A, Meyer J, editor. Springer; 2006. Biomedical Information Visualization; pp. 311&#x2013;342.http://dx.doi.org/10.1007/978-3-540-71949-6_8</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1007/978-3-540-71949-6_8</ArticleId></ArticleIdList></Reference><Reference><Citation>Schneider TD, Stephens RM. Sequence Logos: A New Way to Display Consensus Sequences. Nucleic Acids Research. 1990;18:6097&#x2013;6100. doi: 10.1093/nar/18.20.6097.</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1093/nar/18.20.6097</ArticleId><ArticleId IdType=\"pmc\">PMC332411</ArticleId><ArticleId IdType=\"pubmed\">2172928</ArticleId></ArticleIdList></Reference><Reference><Citation>Slack J, Hildebrand K, Munzner T, John KS. In: Proceedings of the German Conference on Bioinformatics. Giegerich R, Stoye J, editor. Vol. 53. 2004. SequenceJuxtaposer: Fluid Navigation For Large-Scale Sequence Comparison in Context; pp. 37&#x2013;42.</Citation></Reference><Reference><Citation>Mayor C, Brudno M, Schwartz JR, Poliakov A, Rubin EM, Frazer KA, Pachter LS, Dubchak I. VISTA: Visualizing Global DNA Sequence Alignments of Arbitrary Length. Bioinformatics. 2000;16(11):1046&#x2013;1047. doi: 10.1093/bioinformatics/16.11.1046.</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1093/bioinformatics/16.11.1046</ArticleId><ArticleId IdType=\"pubmed\">11159318</ArticleId></ArticleIdList></Reference><Reference><Citation>Shah N, Couronne O, Pennacchio LA, Brudno M, Batzoglou S, Bethel EW, Rubin EM, Hamann B, Dubchak I. Phylo-VISTA: Interactive visualization of multiple DNA sequence alignments. Bioinformatics. 2004;20(5):636&#x2013;643. doi: 10.1093/bioinformatics/btg459.</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1093/bioinformatics/btg459</ArticleId><ArticleId IdType=\"pubmed\">15033870</ArticleId></ArticleIdList></Reference><Reference><Citation>Spell R, Brady R, Dietrich F. BARD: A visualization tool for biological sequence analysis. IEEE Symposium on Information Visualization 2003. 2003. pp. 219&#x2013;225.</Citation></Reference><Reference><Citation>InterPro. http://www.ebi.ac.uk/interpro/</Citation></Reference><Reference><Citation>Hunter S, Apweiler R, Attwood TK, Bairoch A, Bateman A, Binns D, Bork P, Das U, Daugherty L, Duquenne L, Finn RD, Gough J, Haft D, Hulo N, Kahn D, Kelly E, Laugraud A, Letunic I, Lonsdale D, Lopez R, Madera M, Maslen J, McAnulla C, McDowall J, Mistry J, Mitchell A, Mulder N, Natale D, Orengo C, Quinn AF, Selengut JD, Sigrist CJA, Thimma M, Thomas PD, Valentin F, Wilson D, Wu CH, Yeats C. InterPro: The integrative protein signature database. Nucleic Acids Research. 2009. pp. D211&#x2013;D215.</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1093/nar/gkn785</ArticleId><ArticleId IdType=\"pmc\">PMC2686546</ArticleId><ArticleId IdType=\"pubmed\">18940856</ArticleId></ArticleIdList></Reference><Reference><Citation>Consortium TU. The Universal Protein Resource (UniProt) Nucleic Acids Research. 2008. pp. D190&#x2013;D195.</Citation><ArticleIdList><ArticleId IdType=\"pmc\">PMC2238893</ArticleId><ArticleId IdType=\"pubmed\">18045787</ArticleId></ArticleIdList></Reference><Reference><Citation>UniProt. http://www.uniprot.org/</Citation></Reference><Reference><Citation>Clamp M, Cuff J, Searle S, Barton G. The Jalview Java alignment editor. Bioinformatics. 2004;20(3):426&#x2013;427. doi: 10.1093/bioinformatics/btg430.</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1093/bioinformatics/btg430</ArticleId><ArticleId IdType=\"pubmed\">14960472</ArticleId></ArticleIdList></Reference><Reference><Citation>Felsensein J. PHYLIP. http://evolution.genetics.washington.edu/phylip.html</Citation></Reference><Reference><Citation>Henikoff S, Henikoff JG. Amino acid substitution matrices from protein blocks. Proc Natl Acad Sci USA. 1992;89(22):10915&#x2013;10919. doi: 10.1073/pnas.89.22.10915.</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1073/pnas.89.22.10915</ArticleId><ArticleId IdType=\"pmc\">PMC50453</ArticleId><ArticleId IdType=\"pubmed\">1438297</ArticleId></ArticleIdList></Reference><Reference><Citation>McLaren K. The development of the CIE 1976 (L*a*b*) uniform colour-space and colour-difference formula. Journal of the Society of Dyers and Colourists. 1976;92:338&#x2013;341.</Citation></Reference><Reference><Citation>Ware C. Information Visualization: Perception for Design. 2. Morgan Kaufmann; 2004.</Citation></Reference><Reference><Citation>Pirolli P, Card SK. Information Foraging. Psychological Review. 1999;4:643&#x2013;674. doi: 10.1037/0033-295X.106.4.643.</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1037/0033-295X.106.4.643</ArticleId></ArticleIdList></Reference><Reference><Citation>Cock P, Antao T, Chang J, Chapman B, Cox C, Dalke A, Friedberg I, Hamelryck T, Kauff F, Wilczynski B, de Hoon M. Biopython: Freely available Python tools for computational molecular biology and bioinformatics. Bioinformatics. 2009;25(11):1422&#x2013;1423. doi: 10.1093/bioinformatics/btp163.</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1093/bioinformatics/btp163</ArticleId><ArticleId IdType=\"pmc\">PMC2682512</ArticleId><ArticleId IdType=\"pubmed\">19304878</ArticleId></ArticleIdList></Reference><Reference><Citation>PyOpenGL. http://pyopengl.sourceforge.net/</Citation></Reference><Reference><Citation>wxPython. http://wxpython.org/</Citation></Reference><Reference><Citation>McCarthy F, Wang N, Magee G, Nanduri B, Lawrence M, Camon E, Burrell D, Hill D, Dolan M, Williams W, Luthe D, Bridges S, Burgess S. AgBase: A Functional Genomics Resource for Agriculture. BMC Genomics. 2006;7(229)</Citation><ArticleIdList><ArticleId IdType=\"pmc\">PMC1618847</ArticleId><ArticleId IdType=\"pubmed\">16961921</ArticleId></ArticleIdList></Reference><Reference><Citation>MSAVis. http://agbase.msstate.edu/tools/MSAVis.html</Citation></Reference><Reference><Citation>Bailey TL, Boden M, Buske FA, Frith M, Grant CE, Clementi L, Ren J, Li WW, Noble WS. MEME SUITE: Tools for motif discovery and searching. Nucleic Acids Research. 2009. http://dx.doi.org/10.1093/nar/gkp335</Citation><ArticleIdList><ArticleId IdType=\"doi\">10.1093/nar/gkp335</ArticleId><ArticleId IdType=\"pmc\">PMC2703892</ArticleId><ArticleId IdType=\"pubmed\">19458158</ArticleId></ArticleIdList></Reference><Reference><Citation>The MEME Suite. http://meme.nbcr.net/</Citation></Reference><Reference><Citation>Nicholas KB, Nicholas HB, Deerfield DW. GeneDoc: Analysis and visualization of genetic variation. EMBNEW NEWS. 1997;4:14.</Citation></Reference><Reference><Citation>GeneDoc. http://www.nrbsc.org/gfx/genedoc/</Citation></Reference><Reference><Citation>Hall T. BioEdit Sequence Alignment Editor. http://www.mbio.ncsu.edu/BioEdit/bioedit.html</Citation></Reference></ReferenceList></PubmedData></PubmedArticle></PubmedArticleSet>'\n"
     ]
    }
   ],
   "source": [
    "handle = Entrez.efetch(db=\"pubmed\", id=\"19811691\")\n",
    "print(handle.read())\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.2. Nucleotide"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['2727771785', '2727771771', '2727771757', '2727771744', '2727771730', '2727771715', '2727771699', '2727771685', '2727771670', '2727771657']"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "handle = Entrez.esearch(db=\"nucleotide\",retmax=10 ,term=\"Severe acute respiratory syndrome\")\n",
    "record = Entrez.read(handle)\n",
    "record[\"IdList\"]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LOCUS       MW656254               29832 bp    RNA     linear   VRL 23-NOV-2021\n",
      "DEFINITION  Severe acute respiratory syndrome coronavirus 2 isolate\n",
      "            SARS-CoV-2/human/USA/NC-SLPH-0027/2020 ORF1ab polyprotein (ORF1ab),\n",
      "            ORF1a polyprotein (ORF1ab), surface glycoprotein (S), ORF3a protein\n",
      "            (ORF3a), envelope protein (E), membrane glycoprotein (M), and ORF6\n",
      "            protein (ORF6) genes, complete cds; ORF7a protein (ORF7a) and ORF7b\n",
      "            (ORF7b) genes, partial cds; and ORF8 protein (ORF8), nucleocapsid\n",
      "            phosphoprotein (N), and ORF10 protein (ORF10) genes, complete cds.\n",
      "ACCESSION   MW656254\n",
      "VERSION     MW656254.1\n",
      "KEYWORDS    purposeofsampling:baselinesurveillance.\n",
      "SOURCE      Severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2)\n",
      "  ORGANISM  Severe acute respiratory syndrome coronavirus 2\n",
      "            Viruses; Riboviria; Orthornavirae; Pisuviricota; Pisoniviricetes;\n",
      "            Nidovirales; Cornidovirineae; Coronaviridae; Orthocoronavirinae;\n",
      "            Betacoronavirus; Sarbecovirus; Severe acute respiratory\n",
      "            syndrome-related coronavirus.\n",
      "REFERENCE   1  (bases 1 to 29832)\n",
      "  AUTHORS   Chase,K.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (24-FEB-2021) Infectious Diseases, NC SLPH COVID-19\n",
      "            Response Team, 4312 District Dr, Raleigh, NC 27699, USA\n",
      "COMMENT     ##Assembly-Data-START##\n",
      "            Assembly Method       :: EDGE-covid19: BWA 0.7.12-r1039 v. v2.4.0\n",
      "            Sequencing Technology :: Illumina\n",
      "            ##Assembly-Data-END##\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..29832\n",
      "                     /organism=\"Severe acute respiratory syndrome coronavirus\n",
      "                     2\"\n",
      "                     /mol_type=\"genomic RNA\"\n",
      "                     /isolate=\"SARS-CoV-2/human/USA/NC-SLPH-0027/2020\"\n",
      "                     /host=\"Homo sapiens\"\n",
      "                     /db_xref=\"taxon:2697049\"\n",
      "                     /country=\"USA: North Carolina\"\n",
      "                     /collection_date=\"2020-09-11\"\n",
      "     gene            229..21518\n",
      "                     /gene=\"ORF1ab\"\n",
      "     CDS             join(229..13431,13431..21518)\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /ribosomal_slippage\n",
      "                     /codon_start=1\n",
      "                     /product=\"ORF1ab polyprotein\"\n",
      "                     /protein_id=\"QSB46696.1\"\n",
      "                     /translation=\"MESLVPGFNEKTHVQLSLPVLQVRDVLVRGFGDSVEEVLSEARQ\n",
      "                     HLKDGTCGLVEVEKGVLPQLEQPYVFIKRSDARTAPHGHVMVELVAELEGIQYGRSGE\n",
      "                     TLGVLVPHVGEIPVAYRKVLLRKNGNKGAGGHSYGADLKSFDLGDELGTDPYEDFQEN\n",
      "                     WNTKHSSGVTRELMRELNGGAYTRYVDNNFCGPDGYPLECIKDLLARAGKASCTLSEQ\n",
      "                     LDFIDTKRGVYCCREHEHEIAWYTERSEKSYELQTPFEIKLAKKFDIFNGECPNFVFP\n",
      "                     LNSIIKTIQPRVEKKKLDGFMGRIRSVYPVASPNECNQMCLSTLMKCDHCGETSWQTG\n",
      "                     DFVKATCEFCGTENLTKEGATTCGYLPQNAVVKIYCPACHNSEVGPEHSLAEYHNESG\n",
      "                     LKTILRKGGRTIAFGGCVFSYVGCHNKCAYWVPRASANIGCNHTGVVGEGSEGLNDNL\n",
      "                     LEILQKEKVNINIVGDFKLNEEIAIILASFSASTSAFVETVKGLDYKAFKQIVESCGN\n",
      "                     FKVTKGKAKKGAWNIGEQKSILSPLYAFASEAARVVRSIFSRTLETAQNSVRVLQKAA\n",
      "                     ITILDGISQYSLRLIDAMMFTSDLATNNLVVMAYITGGVVQLTSQWLTNIFGTVYEKL\n",
      "                     KPVLDWLEEKFKEGVEFLRDGWEIVKFISTCACEIVGGQIVTCAKEIKESVQTFFKLV\n",
      "                     NKFLALCADSIIIGGAKLKALNLGETFVTHSKGLYRKCVKSREETGLLMPLKAPKEII\n",
      "                     FLEGETLPTEVLTEEVVLKTGDLQPLEQPTSEAVEAPLVGTPVCINGLMLLEIKDTEK\n",
      "                     YCALAPNMMVTNNTFTLKGGAPTKVTFGDDTVIEXQGYKSVNITFELDERIDKVLNEK\n",
      "                     CSAYTVELGTEVNEFACVVADAVIKTLQPVSELLTPLGIDLDEWSMATYYLFDESGEF\n",
      "                     KLASHMYCSFYPPDEDEEEGDCEEEEFEPSTQYEYGTEDDYQGKPLEFGATSAALQPE\n",
      "                     EEQEEDWLDDDSQQTVGQQDGSEDNQTTTIQTIVEVQPQLEMELTPVVQTIEVNSFSG\n",
      "                     YLKLTDNVYIKNADIVEEAKKVKPTVVVNAANVYLKHGGGVAGALNKATNNAMQVESD\n",
      "                     DYIATNGPLKVGGSCVLSGHNLAKHCLHVVGPNVNKGEDIQLLKSAYENFNQHEVLLA\n",
      "                     PLLSAGIFGADPIHSLRVCVDTVRTNVYLAVFDKNLYDKLVSSFLEMKSEKQVEQKIA\n",
      "                     EIPKEEVKPFITESKPSVEQRKQDDKKIKACVEEVTTTLEETKFLTENLLLYIDINGN\n",
      "                     LHPDSATLVSDIDITFLKKDAPYIVGDVVQEGVLTAVVIPTKKAGGTTEMLAKALRKV\n",
      "                     PTDNYITTYPGQGLNGYTVEEAKTVLKKCKSAFYILPSIISNEKQEILGTVSWNLREM\n",
      "                     LAHAEETRKLMPVCVETKAIVSTIQRKYKGIKIQEGVVDYGARFYFYTSKTTVASLIN\n",
      "                     TLNDLNETLVTMPLGYVTHGLNLEEAARYMRSLKVPATVSVSSPDAVTAYNGYLTSSS\n",
      "                     KTPEEHFIETISLAGSYKDWSYSGQSTQLGIEFLKRGDKSVYYTSNPTTFHLDGEVIT\n",
      "                     FDNLKTLLSLREVRTIKVFTTVDNINLHTQVVDMSMTYGQQFGPTYLDGADVTKIKPH\n",
      "                     NSHEGKTFYVLPNDDTLRVEAFEYYHTTDPSFLGRYMSALNHTKKWKYPQVNGLTSIK\n",
      "                     WADNNCYLATALLTLQQIELKFNPPALQDAYYRARAGEAANFCALILAYCNKTVGELG\n",
      "                     DVRETMSYLFQHANLDSCKRVLNVVCKTCGQQQTTLKGVEAVMYMGTLSYEQFKKGVQ\n",
      "                     IPCTCGKQATKYLVQQESPFVMMSAPPAQYELKHGTFTCASEYTGNYQCGHYKHITSK\n",
      "                     ETLYCIDGALLTKSSEYKGPITDVFYKENSYTTTIKPVTYKLDGVVCTEIDPKLDNYY\n",
      "                     KKDNSYFTEQPIDLVPNQPYPNASFDNFKFVCDNIKFADDLNQLTGYKKPASRELKVT\n",
      "                     FFPDLNGDVVAIDYKHYTPSFKKGAKLLHKPIVWHVNNATNKATYKPNTWCIRCLWST\n",
      "                     KPVETSNSFDVLKSEDAQGMDNLACEDLKPVSEEVVENPTIQKDVLECNVKTTEVVGD\n",
      "                     IILKPANNSLKIIEEVGHTDLMAAYVDNSSLTIKKPNELSRVLGLKTLATHGLAAVNS\n",
      "                     VPWDTIANYAKPFLNKVVSTTTNIVTRCLNRVCTNYMPYFFTLLLQLCTFTRSTNSRI\n",
      "                     KASMPTTIAXNTVKSVGKFCLEASFNYLKSPNFSKLINIIIWFLLLSVCLGSLIYSTA\n",
      "                     ALGVLMSNLGMPSYCTGYREXYLNSTNVTIATYCTGSIPCSVCLSGLDSLDTYPSLET\n",
      "                     IQITISSFKWDLTAFGLVAEWFLAYILFTRFFYVLGLAAIMQLFFSYFAVHFISNSWL\n",
      "                     MWLIINLVQMAPISAMVRMYIFFASFYYVWKSYVHVVDGCNSSTCMMCYKRNRATRVE\n",
      "                     CTTIVNGVRRSFYVYANGGKGFCKLHNWNCVNCDTFCAGSTFISDEVARDLSLQFKRP\n",
      "                     INPTDQSSYIVDSVTVKNGSIHLYFDKAGQKTYERHSLSHFVNLDNLRANNTKGSLPI\n",
      "                     NVIVFDGKSKCEESSAKSASVYYSQLMCQPILLLDQALVSDVGDSAEVAVKMFDAYVN\n",
      "                     TFSSTFNVPIEKLKTLVATAEAELAKNVSLDNVLSTFISAARQGFVDSDVETKDVVEC\n",
      "                     LKLSHQSDIEVTGDSCNNYMLTYNKVENMTPRDLGACIDCSARHINAQVAKSHNIALI\n",
      "                     WNVKDFMSLSEQLRKQIRSAAKKNNLPFKLTCATTRQVVNVVTTKIALKGGKIVNNWL\n",
      "                     KQLIKVTLVFLFVAAIFYLITPVHVMSKHTDFSSEIIGYKAIDGGVTRDIASTDTCFA\n",
      "                     NKHADFDTWFSQRGGSYTNDKACPLIAAVITREVGFVVPGLPGTILRTTNGDFLHFLP\n",
      "                     RVFSAVGNICYTPSKLIEYTDFATSACVLAAECTIFKDASGKPVPYCYDTNVLEGSVA\n",
      "                     YESLRPDTRYVLMDGSIIQFPNTYLEGSVRVVTTFDSEYCRHGTCERSEAGVCVSTSG\n",
      "                     RWVLNNDYYRSLPGVFCGVDAVNLLTNMFTPLIQPIGALDISASIVAGGIVAIVVTCL\n",
      "                     AYYFMRFRRAFGEYSHVVAFNTLLFLMSFTVLCLTPVYSFLPGVYSVIYLYLTFYLTN\n",
      "                     DVSFLAHIQWMVMFTPLVPFWITIAYIICISTKHFYWFFSNYLKRRVVFNGVSFSTFE\n",
      "                     EAALCTFLLNKEMYLKLRSDVLLPLTQYNRYLALYNKYKYFSGAMDTTSYREAACCHL\n",
      "                     AKALNDFSNSGSDVLYQPPQTSITSAVLQSGFRKMAFPSGKVEGCMVQVTCGTTTLNG\n",
      "                     LWLDDVVYCPRHVICTSEDMLNPNYEDLLIRKSNHNFLVQAGNVQLRVIGHSMQNCVL\n",
      "                     KFKVDTANPKTPKYKFVRIQPGQTFSVLACYNGSPSGVYQCAMRPNFTIKGSFLNGSC\n",
      "                     GSVGFNIDYDCVSFCYMHHMELPTGVHAGTDLEGNFYGPFVDRQTAQAAGTDTTITVN\n",
      "                     VLAWLYAAVINGDRWFLNRFTTTLNDFNLVAMKYNYEPLTQDHVDILGPLSAQTGIAV\n",
      "                     LDMCASLKELLQNGMNGRTILGSALLEDEFTPFDVVRQCSGVTFQSAVKRTIKGTHHW\n",
      "                     LLLTILTSLLVLVQSTQWSLFFFLYENAFLPFAMGIIAMSAFAMMFVKHKHAFLCLFL\n",
      "                     LPSLATVAYFNMVYMPASWVMRIMTWLDMVDTSLSGFKLKDCVMYASAVVLLILMTAR\n",
      "                     TVYDDGARRVWTLMNVLTLVYKVYYGNALDQAISMWALIISVTSNYSGVVTTVMFLAR\n",
      "                     GIVFMCVEYCPIFFITGNTLQCIMLVYCFLGYFCTCYFGLFCLLNRYFRLTLGVYDYL\n",
      "                     VSTQEFRYMNSQGLLPPKNSIDAFKLNIKLLGVGGKPCIKVATVQSKMSDVKCTSVVL\n",
      "                     LSVLQQLRVESSSKLWAQCVQLHNDILLAKDTTEAFEKMVSLLSVLLSMQGAVDINKL\n",
      "                     CEEMLDNRATLQAIASEFSSLPSYAAFATAQEAYEQAVANGDSEVVLKKLKKSLNVAK\n",
      "                     SEFDRDAAMQRKLEKMADQAMTQMYKQARSEDKRAKVTSAMQTMLFTMLRKLDNDALN\n",
      "                     NIINNARDGCVPLNIIPLTTAAKLMVVIPDYNTYKNTCDGTTFTYASALWEIQQVVDA\n",
      "                     DSKIVQLSEISMDNSPNLAWPLIVTALRANSAVKLQNNELSPVALRQMSCAAGTTQTA\n",
      "                     CTDDNALAYYNTTKGGRFVLALLSDLQDLKWARFPKSDGTGTIYTELEPPCRFVTDTP\n",
      "                     KGPKVKYLYFIKGLNNLNRGMVLGSLAATVRLQAGNATEVPANSTVLSFCAFAVDAAK\n",
      "                     AYKDYLASGGQPITNCVKMLCTHTGTGQAITVTPEANMDQESFGGASCCLYCRCHIDH\n",
      "                     PNPKGFCDLKGKYVQIPTTCANDPVGFTLKNTVCTVCGMWKGYGCSCDQLREPMLQSA\n",
      "                     DAQSFLNRVCGVSAARLTPCGTGTSTDVVYRAFDIYNDKVAGFAKFLKTNCCRFQEKD\n",
      "                     EDDNLIDSYFVVKRHTFSNYQHEETIYNLLKDCPAVAKHDFFKFRIDGDMVPHISRQR\n",
      "                     LTKYTMADLVYALRHFDEGNCDTLKEILVTYNCCDDDYFNKKDWYDFVENPDILRVYA\n",
      "                     NLGERVRQALLKTVQFCDAMRNAGIVGVLTLDNQDLNGNWYDFGDFIQTTPGSGVPVV\n",
      "                     DSYYSLLMPILTLTRALTAESHVDTDLTKPYIKWDLLKYDFTEERLKLFDRYFKYWDQ\n",
      "                     TYHPNCVNCLDDRCILHCANFNVLFSTVFPLTSFGPLVRKIFVDGVPFVVSTGYHFRE\n",
      "                     LGVVHNQDVNLHSSRLSFKELLVYAADPAMHAASGNLLLDKRTTCFSVAALTNNVAFQ\n",
      "                     TVKPGNFNKDFYDFAVSKGFFKEGSSVELKHFFFAQDGNAAISDYDYYRYNLPTMCDI\n",
      "                     RQLLFVVEVVDKYFDCYDGGCINANQVIVNNLDKSAGFPFNKWGKARLYYDSMSYEDQ\n",
      "                     DALFAYTKRNVIPTITQMNLKYAISAKNRARTVAGVSICSTMTNRQFHQKLLKSIAAT\n",
      "                     RGATVVIGTSKFYGGWHNMLKTVYSDVENPHLMGWDYPKCDRAMPNMLRIMASLVLAR\n",
      "                     KHTTCCSLSHRFYRLANECAQVLSEMVMCGGSLYVKPGGTSSGDATTAYANSVFNICQ\n",
      "                     AVTANVNALLSTDGNKIADKYVRNLQHRLYECLYRNRDVDTDFVNEFYAYLRKHFSMM\n",
      "                     ILSDDAVVCFNSTYASQGLVASIKNFKSVLYYQNNVFMSEAKCWTETDLTKGPHEFCS\n",
      "                     QHTMLVKQGDDYVYLPYPDPSRILGAGCFVDDIVKTDGTLMIERFVSLAIDAYPLTKH\n",
      "                     PNQEYADVFHLYLQYIRKLHDELTGHMLDMYSVMLTNDNTSRYWEPEFYEAMYTPHTV\n",
      "                     LQAVGACVLCNSQTSLRCGACIRRPFLCCKCCYDHVISTSHKLVLSVNPYVCNAPGCD\n",
      "                     VTDVTQLYLGGMSYYCKSHKPPISFPLCANGQVFGLYKNTCVGSDNVTDFNAIATCDW\n",
      "                     TNAGDYILANTCTERLKLFAAETLKATEETFKLSYGIATVREVLSDRELHLSWEVGKP\n",
      "                     RPPLNRNYVFTGYRVTKNSKVQIGEYTFEKGDYGDAVVYRGTTTYKLNVGDYFVLTSH\n",
      "                     TVMPLSAPTLVPQEHYVRITGLYPTLNISDEFSSNVANYQKVGMQKYSTLQGPPGTGK\n",
      "                     SHFAIGLALYYPSARIVYTACSHAAVDALCEKALKYLPIDKCSRIIPARARVECFDKF\n",
      "                     KVNSTLEQYVFCTVNALPETTADIVVFDEISMATNYDLSVVNARLRAKHYVYIGDPAQ\n",
      "                     LPAPRTLLTKGTLEPEYFNSVCRLMKTIGPDMFLGTCRRCPAEIVDTVSALVYDNKLK\n",
      "                     AHKDKSAQCFKMFYKGVITHDVSSAINRPQIGVVREFLTRNPAWRKAVFISPYNSQNA\n",
      "                     VASKILGLPTQTVDSSQGSEYDYVIFTQTTETAHSCNVNRFNVAITRAKVGILCIMSD\n",
      "                     RDLYDKLQFTSLEIPRRNVATLQAENVTGLFKDCSKVITGLHPTQAPTHLSVDTKFKT\n",
      "                     EGLCVDIPGIPKDMTYRRLISMMGFKMNYQVNGYPNMFITREEAIRHVRAWIGFDVEG\n",
      "                     CHATREAVGTNLPLQLGFSTGVNLVAVPTGYVDTPDNTDFSRVSAKPPPGDQFKHLIP\n",
      "                     LMYKGLPWNVVRIKIVQMLSDTLKNLSDRVVFVLWAHGFELTSMKYFVKIGPERTCCL\n",
      "                     CDRRATCFSTASDTYACWHHSIGFDYVYNPFMIDVQQWGFTGNLQSNHDLYCQVHGNA\n",
      "                     HVASCDAIMTRCLAVHECFVKRVDWTIEYPIIGDELKINAACRKVQHMVVKAALLADK\n",
      "                     FPVLHDIGNPKAIKCVPQADVEWKFYDAQPCSDKAYKIEELFYSYATHSDKFTDGVCL\n",
      "                     FWNCNVDRYPANSIVCRFDTRVLSNLNLPGCDGGXXXXXXXXXXXXXXXXXXXXXXXX\n",
      "                     XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\n",
      "                     XXXXXXFSLWVYKQFDTYNLWNTFTRLQSLENVAFNVVNKGHFDGQQGEVPVSIINNT\n",
      "                     VYTKVDGVDVELFENKTTLPVNVAFELWAKRNIKPVPEVKILNNLGVDIAANTVIWDY\n",
      "                     KRDAPAHISTIGVCSMTDIAKKPTETICAPLTVFFDGRVDGQVDLFRNARNGVLITEG\n",
      "                     SVKGLQPSVGPKQASLNGVTLIGEAVKTQFNYYKKVDGVVQQLPETYFTQSRNLQEFK\n",
      "                     PRSQMEIDFLELAMDEFIERYKLEGYAFEHIVYGDFSHSQLGGLHLLIGLAKRFKESP\n",
      "                     FELEDFIPMDSTVKNYFITDAQTGSSKCVCSVIDLLLDDFVEIIKSQDLSVVSKVVKV\n",
      "                     TIDYTEISFMLWCKDGHVETFYPKLQSSQAWQPGVAMPNLYKMQRMLLEKCDLQNYGD\n",
      "                     SATLPKGIMMNVAKYTQLCQYLNTLTLAVPYNMRVIHFGAGSDKGVAPGTAVLRQWLP\n",
      "                     TGTLLVDSDLNDFVSDADSTLIGDCATVHTANKWDLIISDMYDPKTKNVTKENDSKEG\n",
      "                     FFTYICGFIQQKLALGGSVAIKITEHSWNADLYKLMGHFAWWTAFVTNVNASSSEAFL\n",
      "                     XGCNYLGKPCXQIDGYVMHANYIFWRNTNPIQLSSYSLFDMSKFPLKLRGTAVMSLKE\n",
      "                     GQINDMILSLLSKGRLIIRENNRVVISSDVLVNN\"\n",
      "     mat_peptide     229..768\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"leader protein\"\n",
      "     mat_peptide     769..2682\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp2\"\n",
      "     mat_peptide     2683..8517\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp3\"\n",
      "     mat_peptide     8518..10017\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp4\"\n",
      "     mat_peptide     10018..10935\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"3C-like proteinase\"\n",
      "     mat_peptide     10936..11805\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp6\"\n",
      "     mat_peptide     11806..12054\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp7\"\n",
      "     mat_peptide     12055..12648\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp8\"\n",
      "     mat_peptide     12649..12987\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp9\"\n",
      "     mat_peptide     12988..13404\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp10\"\n",
      "     mat_peptide     join(13405..13431,13431..16199)\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"RNA-dependent RNA polymerase\"\n",
      "     mat_peptide     16200..18002\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"helicase\"\n",
      "     mat_peptide     18003..19583\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"3'-to-5' exonuclease\"\n",
      "     mat_peptide     19584..20621\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"endoRNAse\"\n",
      "     mat_peptide     20622..21515\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"2'-O-ribose methyltransferase\"\n",
      "     CDS             229..13446\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"ORF1a polyprotein\"\n",
      "                     /protein_id=\"QSB46697.1\"\n",
      "                     /translation=\"MESLVPGFNEKTHVQLSLPVLQVRDVLVRGFGDSVEEVLSEARQ\n",
      "                     HLKDGTCGLVEVEKGVLPQLEQPYVFIKRSDARTAPHGHVMVELVAELEGIQYGRSGE\n",
      "                     TLGVLVPHVGEIPVAYRKVLLRKNGNKGAGGHSYGADLKSFDLGDELGTDPYEDFQEN\n",
      "                     WNTKHSSGVTRELMRELNGGAYTRYVDNNFCGPDGYPLECIKDLLARAGKASCTLSEQ\n",
      "                     LDFIDTKRGVYCCREHEHEIAWYTERSEKSYELQTPFEIKLAKKFDIFNGECPNFVFP\n",
      "                     LNSIIKTIQPRVEKKKLDGFMGRIRSVYPVASPNECNQMCLSTLMKCDHCGETSWQTG\n",
      "                     DFVKATCEFCGTENLTKEGATTCGYLPQNAVVKIYCPACHNSEVGPEHSLAEYHNESG\n",
      "                     LKTILRKGGRTIAFGGCVFSYVGCHNKCAYWVPRASANIGCNHTGVVGEGSEGLNDNL\n",
      "                     LEILQKEKVNINIVGDFKLNEEIAIILASFSASTSAFVETVKGLDYKAFKQIVESCGN\n",
      "                     FKVTKGKAKKGAWNIGEQKSILSPLYAFASEAARVVRSIFSRTLETAQNSVRVLQKAA\n",
      "                     ITILDGISQYSLRLIDAMMFTSDLATNNLVVMAYITGGVVQLTSQWLTNIFGTVYEKL\n",
      "                     KPVLDWLEEKFKEGVEFLRDGWEIVKFISTCACEIVGGQIVTCAKEIKESVQTFFKLV\n",
      "                     NKFLALCADSIIIGGAKLKALNLGETFVTHSKGLYRKCVKSREETGLLMPLKAPKEII\n",
      "                     FLEGETLPTEVLTEEVVLKTGDLQPLEQPTSEAVEAPLVGTPVCINGLMLLEIKDTEK\n",
      "                     YCALAPNMMVTNNTFTLKGGAPTKVTFGDDTVIEXQGYKSVNITFELDERIDKVLNEK\n",
      "                     CSAYTVELGTEVNEFACVVADAVIKTLQPVSELLTPLGIDLDEWSMATYYLFDESGEF\n",
      "                     KLASHMYCSFYPPDEDEEEGDCEEEEFEPSTQYEYGTEDDYQGKPLEFGATSAALQPE\n",
      "                     EEQEEDWLDDDSQQTVGQQDGSEDNQTTTIQTIVEVQPQLEMELTPVVQTIEVNSFSG\n",
      "                     YLKLTDNVYIKNADIVEEAKKVKPTVVVNAANVYLKHGGGVAGALNKATNNAMQVESD\n",
      "                     DYIATNGPLKVGGSCVLSGHNLAKHCLHVVGPNVNKGEDIQLLKSAYENFNQHEVLLA\n",
      "                     PLLSAGIFGADPIHSLRVCVDTVRTNVYLAVFDKNLYDKLVSSFLEMKSEKQVEQKIA\n",
      "                     EIPKEEVKPFITESKPSVEQRKQDDKKIKACVEEVTTTLEETKFLTENLLLYIDINGN\n",
      "                     LHPDSATLVSDIDITFLKKDAPYIVGDVVQEGVLTAVVIPTKKAGGTTEMLAKALRKV\n",
      "                     PTDNYITTYPGQGLNGYTVEEAKTVLKKCKSAFYILPSIISNEKQEILGTVSWNLREM\n",
      "                     LAHAEETRKLMPVCVETKAIVSTIQRKYKGIKIQEGVVDYGARFYFYTSKTTVASLIN\n",
      "                     TLNDLNETLVTMPLGYVTHGLNLEEAARYMRSLKVPATVSVSSPDAVTAYNGYLTSSS\n",
      "                     KTPEEHFIETISLAGSYKDWSYSGQSTQLGIEFLKRGDKSVYYTSNPTTFHLDGEVIT\n",
      "                     FDNLKTLLSLREVRTIKVFTTVDNINLHTQVVDMSMTYGQQFGPTYLDGADVTKIKPH\n",
      "                     NSHEGKTFYVLPNDDTLRVEAFEYYHTTDPSFLGRYMSALNHTKKWKYPQVNGLTSIK\n",
      "                     WADNNCYLATALLTLQQIELKFNPPALQDAYYRARAGEAANFCALILAYCNKTVGELG\n",
      "                     DVRETMSYLFQHANLDSCKRVLNVVCKTCGQQQTTLKGVEAVMYMGTLSYEQFKKGVQ\n",
      "                     IPCTCGKQATKYLVQQESPFVMMSAPPAQYELKHGTFTCASEYTGNYQCGHYKHITSK\n",
      "                     ETLYCIDGALLTKSSEYKGPITDVFYKENSYTTTIKPVTYKLDGVVCTEIDPKLDNYY\n",
      "                     KKDNSYFTEQPIDLVPNQPYPNASFDNFKFVCDNIKFADDLNQLTGYKKPASRELKVT\n",
      "                     FFPDLNGDVVAIDYKHYTPSFKKGAKLLHKPIVWHVNNATNKATYKPNTWCIRCLWST\n",
      "                     KPVETSNSFDVLKSEDAQGMDNLACEDLKPVSEEVVENPTIQKDVLECNVKTTEVVGD\n",
      "                     IILKPANNSLKIIEEVGHTDLMAAYVDNSSLTIKKPNELSRVLGLKTLATHGLAAVNS\n",
      "                     VPWDTIANYAKPFLNKVVSTTTNIVTRCLNRVCTNYMPYFFTLLLQLCTFTRSTNSRI\n",
      "                     KASMPTTIAXNTVKSVGKFCLEASFNYLKSPNFSKLINIIIWFLLLSVCLGSLIYSTA\n",
      "                     ALGVLMSNLGMPSYCTGYREXYLNSTNVTIATYCTGSIPCSVCLSGLDSLDTYPSLET\n",
      "                     IQITISSFKWDLTAFGLVAEWFLAYILFTRFFYVLGLAAIMQLFFSYFAVHFISNSWL\n",
      "                     MWLIINLVQMAPISAMVRMYIFFASFYYVWKSYVHVVDGCNSSTCMMCYKRNRATRVE\n",
      "                     CTTIVNGVRRSFYVYANGGKGFCKLHNWNCVNCDTFCAGSTFISDEVARDLSLQFKRP\n",
      "                     INPTDQSSYIVDSVTVKNGSIHLYFDKAGQKTYERHSLSHFVNLDNLRANNTKGSLPI\n",
      "                     NVIVFDGKSKCEESSAKSASVYYSQLMCQPILLLDQALVSDVGDSAEVAVKMFDAYVN\n",
      "                     TFSSTFNVPIEKLKTLVATAEAELAKNVSLDNVLSTFISAARQGFVDSDVETKDVVEC\n",
      "                     LKLSHQSDIEVTGDSCNNYMLTYNKVENMTPRDLGACIDCSARHINAQVAKSHNIALI\n",
      "                     WNVKDFMSLSEQLRKQIRSAAKKNNLPFKLTCATTRQVVNVVTTKIALKGGKIVNNWL\n",
      "                     KQLIKVTLVFLFVAAIFYLITPVHVMSKHTDFSSEIIGYKAIDGGVTRDIASTDTCFA\n",
      "                     NKHADFDTWFSQRGGSYTNDKACPLIAAVITREVGFVVPGLPGTILRTTNGDFLHFLP\n",
      "                     RVFSAVGNICYTPSKLIEYTDFATSACVLAAECTIFKDASGKPVPYCYDTNVLEGSVA\n",
      "                     YESLRPDTRYVLMDGSIIQFPNTYLEGSVRVVTTFDSEYCRHGTCERSEAGVCVSTSG\n",
      "                     RWVLNNDYYRSLPGVFCGVDAVNLLTNMFTPLIQPIGALDISASIVAGGIVAIVVTCL\n",
      "                     AYYFMRFRRAFGEYSHVVAFNTLLFLMSFTVLCLTPVYSFLPGVYSVIYLYLTFYLTN\n",
      "                     DVSFLAHIQWMVMFTPLVPFWITIAYIICISTKHFYWFFSNYLKRRVVFNGVSFSTFE\n",
      "                     EAALCTFLLNKEMYLKLRSDVLLPLTQYNRYLALYNKYKYFSGAMDTTSYREAACCHL\n",
      "                     AKALNDFSNSGSDVLYQPPQTSITSAVLQSGFRKMAFPSGKVEGCMVQVTCGTTTLNG\n",
      "                     LWLDDVVYCPRHVICTSEDMLNPNYEDLLIRKSNHNFLVQAGNVQLRVIGHSMQNCVL\n",
      "                     KFKVDTANPKTPKYKFVRIQPGQTFSVLACYNGSPSGVYQCAMRPNFTIKGSFLNGSC\n",
      "                     GSVGFNIDYDCVSFCYMHHMELPTGVHAGTDLEGNFYGPFVDRQTAQAAGTDTTITVN\n",
      "                     VLAWLYAAVINGDRWFLNRFTTTLNDFNLVAMKYNYEPLTQDHVDILGPLSAQTGIAV\n",
      "                     LDMCASLKELLQNGMNGRTILGSALLEDEFTPFDVVRQCSGVTFQSAVKRTIKGTHHW\n",
      "                     LLLTILTSLLVLVQSTQWSLFFFLYENAFLPFAMGIIAMSAFAMMFVKHKHAFLCLFL\n",
      "                     LPSLATVAYFNMVYMPASWVMRIMTWLDMVDTSLSGFKLKDCVMYASAVVLLILMTAR\n",
      "                     TVYDDGARRVWTLMNVLTLVYKVYYGNALDQAISMWALIISVTSNYSGVVTTVMFLAR\n",
      "                     GIVFMCVEYCPIFFITGNTLQCIMLVYCFLGYFCTCYFGLFCLLNRYFRLTLGVYDYL\n",
      "                     VSTQEFRYMNSQGLLPPKNSIDAFKLNIKLLGVGGKPCIKVATVQSKMSDVKCTSVVL\n",
      "                     LSVLQQLRVESSSKLWAQCVQLHNDILLAKDTTEAFEKMVSLLSVLLSMQGAVDINKL\n",
      "                     CEEMLDNRATLQAIASEFSSLPSYAAFATAQEAYEQAVANGDSEVVLKKLKKSLNVAK\n",
      "                     SEFDRDAAMQRKLEKMADQAMTQMYKQARSEDKRAKVTSAMQTMLFTMLRKLDNDALN\n",
      "                     NIINNARDGCVPLNIIPLTTAAKLMVVIPDYNTYKNTCDGTTFTYASALWEIQQVVDA\n",
      "                     DSKIVQLSEISMDNSPNLAWPLIVTALRANSAVKLQNNELSPVALRQMSCAAGTTQTA\n",
      "                     CTDDNALAYYNTTKGGRFVLALLSDLQDLKWARFPKSDGTGTIYTELEPPCRFVTDTP\n",
      "                     KGPKVKYLYFIKGLNNLNRGMVLGSLAATVRLQAGNATEVPANSTVLSFCAFAVDAAK\n",
      "                     AYKDYLASGGQPITNCVKMLCTHTGTGQAITVTPEANMDQESFGGASCCLYCRCHIDH\n",
      "                     PNPKGFCDLKGKYVQIPTTCANDPVGFTLKNTVCTVCGMWKGYGCSCDQLREPMLQSA\n",
      "                     DAQSFLNGFAV\"\n",
      "     mat_peptide     229..768\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"leader protein\"\n",
      "     mat_peptide     769..2682\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp2\"\n",
      "     mat_peptide     2683..8517\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp3\"\n",
      "     mat_peptide     8518..10017\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp4\"\n",
      "     mat_peptide     10018..10935\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"3C-like proteinase\"\n",
      "     mat_peptide     10936..11805\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp6\"\n",
      "     mat_peptide     11806..12054\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp7\"\n",
      "     mat_peptide     12055..12648\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp8\"\n",
      "     mat_peptide     12649..12987\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp9\"\n",
      "     mat_peptide     12988..13404\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp10\"\n",
      "     mat_peptide     13405..13443\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /product=\"nsp11\"\n",
      "     stem_loop       13439..13466\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /note=\"Coronavirus frameshifting stimulation element\n",
      "                     stem-loop 1\"\n",
      "     stem_loop       13451..13505\n",
      "                     /gene=\"ORF1ab\"\n",
      "                     /note=\"Coronavirus frameshifting stimulation element\n",
      "                     stem-loop 2\"\n",
      "     gap             19256..19516\n",
      "                     /estimated_length=261\n",
      "     gene            21526..25347\n",
      "                     /gene=\"S\"\n",
      "     CDS             21526..25347\n",
      "                     /gene=\"S\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"surface glycoprotein\"\n",
      "                     /protein_id=\"QSB46698.1\"\n",
      "                     /translation=\"MFVFLVLLPLVSSQCVNLTTRTQLPPAYTNSFTRGVYYPDKVFR\n",
      "                     SSVLHSTQDLFLPFFSNVTWFHAIHVSGTNGTKRFDNPVLPFNDGVYFASTEKSNIIR\n",
      "                     GWIFGTTLDSKTQSLLIVNNATNVVIKVCEFQFCNDPFLGVYYHKNNKSWMESEFRVY\n",
      "                     SSANNCTFEYVSQPFLMDLEGKQGNFKNLREFVFKNIDGYFKIYSKHTPINLVRDLPQ\n",
      "                     GFSALEPLVDLPIGINITRFQTLLALHRSYLTPGDSSSGWTAGAAAYYVGYLQPRTFL\n",
      "                     LKYNENGTITDAVDCALDPLSETKCTLKSFTVEKGIYQTSNFRVQPTESIVRFPNITN\n",
      "                     LCPFGEVFNATRFASVYAWNRKRISNCVADYSVLYNSASFSTFKCYGVSPTKLNDLCF\n",
      "                     TNVYADSFVIRGDEVRQIAPGQTGKIADYNYKLPDDFTGCVIAWNSNNLDSKVGGNYN\n",
      "                     YLYRLFRKSNLKPFERDISTEIYQAGSTPCNGVEGFNCYFPLQSYGFQPTNGVGYQPY\n",
      "                     RVVVLSFELLHAPATVCGPKKSTNLVKNKCVNFNFNGLTGTGVLTESNKKFLPFQQFG\n",
      "                     RDIADTTDAVRDPQTLEILDITPCSFGGVSVITPGTNTSNQVAVLYQGVNCTEVPVAI\n",
      "                     HADQLTPTWRVYSTGSNVFQTRAGCLIGAEHVNNSYECDIPIGAGICASYQTQTNSPR\n",
      "                     RARSVASQSIIAYTMSLGAENSVAYSNNSIAIPTNFTISVTTEILPVSMTKTSVDCTM\n",
      "                     YICGDSTECSNLLLQYGSFCTQLNRALTGIAVEQDKNTQEVFAQVKQIYKTPPIKDFG\n",
      "                     GFNFSQILPDPSKPSKRSFIEDLLFNKVTLADAGFIKQYGDCLGDIAARDLICAQKFN\n",
      "                     GLTVLPPLLTDEMIAQYTSALLAGTITSGWTFGAGAALQIPFAMQMAYRFNGIGVTQN\n",
      "                     VLYENQKLIANQFNSAIGKIQDSLSSTASALGKLQDVVNQNAQALNTLVKQLSSNFGA\n",
      "                     ISSVLNDILSRLDKVEAEVQIDRLITGRLQSLQTYVTQQLIRAAEIRASANLAATKMS\n",
      "                     ECVLGQSKRVDFCGKGYHLMSFPQSAPHGVVFLHVTYVPAQEKNFTTAPAICHDGKAH\n",
      "                     FPREGVFVSNGTHWFVTQRNFYEPQIITTDNTFVSGNCDVVIGIVNNTVYDPLQPELD\n",
      "                     SFKEELDKYFKNHTSPDVDLGDISGINASVVNIQKEIDRLNEVAKNLNESLIDLQELG\n",
      "                     KYEQYIKWPWYIWLGFIAGLIAIVMVTIMLCCMTSCCSCLKGCCSCGSCCKFDEDDSE\n",
      "                     PVLKGVKLHYT\"\n",
      "     gene            25356..26183\n",
      "                     /gene=\"ORF3a\"\n",
      "     CDS             25356..26183\n",
      "                     /gene=\"ORF3a\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"ORF3a protein\"\n",
      "                     /protein_id=\"QSB46699.1\"\n",
      "                     /translation=\"MDLFMRIFTIGTVTLKQGEIKDATPSDFVRATATIPIQASLPFG\n",
      "                     WLIVGVALLAVFHSASKIITLKKRWQLALSKGVHFVCNLLLLFVTVYSHLLLVAAGLE\n",
      "                     APFLYLYALVYFLQSINFVRIIMRLWLCWKCRSKNPLLYDANYFLCWHTNCYDYCIPY\n",
      "                     NSVTSSIVITSVDGTTSPISEHDYQIGGYTEKWESGVKDCVVLHSYFTSDYYQLYSTQ\n",
      "                     LSTDTGVEHVTFFIYNKIVDEPEEHVQIHTIDGSSGVVNPVMEPIYDEPTTTTSVPL\"\n",
      "     gene            26208..26435\n",
      "                     /gene=\"E\"\n",
      "     CDS             26208..26435\n",
      "                     /gene=\"E\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"envelope protein\"\n",
      "                     /protein_id=\"QSB46700.1\"\n",
      "                     /translation=\"MYSFVSEETGTLIVNSVLLFLAFVVFLLVTLAILTALRLCAYCC\n",
      "                     NIVNVSLVKPSFYVYSRVKNLNSSRVPDLLV\"\n",
      "     gene            26486..27154\n",
      "                     /gene=\"M\"\n",
      "     CDS             26486..27154\n",
      "                     /gene=\"M\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"membrane glycoprotein\"\n",
      "                     /protein_id=\"QSB46701.1\"\n",
      "                     /translation=\"MADSNGTITVEELKKLLEQWNLVIGFLFLTWICLLQFAYANRNR\n",
      "                     FLYIIKLIFLWLLWPVTLACFVLAAVYRINWITGGIAIAMACLVGLMWLSYFIASFRL\n",
      "                     FARTRSMWSFNPETNILLNVPLHGTILTRPLLESELVIGAVILRGHLRIAGHHLGRCD\n",
      "                     IKDLPKEITVATSRTLSYYKLGASQRVAGDSGFAAYSRYRIGNYKLNTDHSSSSDNIA\n",
      "                     LLVQ\"\n",
      "     gene            27165..27350\n",
      "                     /gene=\"ORF6\"\n",
      "     CDS             27165..27350\n",
      "                     /gene=\"ORF6\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"ORF6 protein\"\n",
      "                     /protein_id=\"QSB46702.1\"\n",
      "                     /translation=\"MFHLVDFQVTIAEILLIIMRTFKVSIWNLDYIINLIIKNLSKSL\n",
      "                     TENKYSQLDEEQPMEID\"\n",
      "     gene            27357..>27680\n",
      "                     /gene=\"ORF7a\"\n",
      "     CDS             27357..>27680\n",
      "                     /gene=\"ORF7a\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"ORF7a protein\"\n",
      "                     /protein_id=\"QSB46703.1\"\n",
      "                     /translation=\"MKIILFLALITLATCELYHYQECVRGTTVLLKEPCSSGTYEGNS\n",
      "                     PFHPLADNXFALTCFSTQFAFXXPXXVKHVYQLRAXXVSPKLFIRQEEVXXLYSPXFL\n",
      "                     IVAAXV\"\n",
      "     gene            <27755..27850\n",
      "                     /gene=\"ORF7b\"\n",
      "     CDS             <27755..27850\n",
      "                     /gene=\"ORF7b\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"ORF7b\"\n",
      "                     /protein_id=\"QSB46704.1\"\n",
      "                     /translation=\"FLAFLLFLVLIMLIIFWFSLELQDHNETCHA\"\n",
      "     gene            27857..28222\n",
      "                     /gene=\"ORF8\"\n",
      "     CDS             27857..28222\n",
      "                     /gene=\"ORF8\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"ORF8 protein\"\n",
      "                     /protein_id=\"QSB46705.1\"\n",
      "                     /translation=\"MKFLVFLGIITTVAAFHQECSLQLCTQHQPYVVDDPCPIHFYSK\n",
      "                     WYIRVGARKSAPLIELCVDEAGSKSPIQYIDIGNYTVSCLPFTINCQEPKLGSLVVRC\n",
      "                     SFYEDFLEYHDVRVVLDFI\"\n",
      "     gene            28237..29496\n",
      "                     /gene=\"N\"\n",
      "     CDS             28237..29496\n",
      "                     /gene=\"N\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"nucleocapsid phosphoprotein\"\n",
      "                     /protein_id=\"QSB46706.1\"\n",
      "                     /translation=\"MSDNGPQNQRNAPRITFGGPSDSTGSNQNGERSGARSKQRRPQG\n",
      "                     LPNNTASWFTALTQHGKEDLKFSRGQGVPINTNSSPDDQIGYYRRATRRIRGGDGKMK\n",
      "                     DLSPRWYFYYLGTGPEAGLPYGANKDGIIWVATEGALNTPKDHIGTRNPANNAAIVLQ\n",
      "                     LPQGTTLPKGFYAEGSRGGSQASSRSSSRSRNSSRNSTLGSSRGTSPARMAGNGGDAA\n",
      "                     LALLLLDRLNQLESKMSGKGQQQQGQTVTKKSAAEASKKPRQKRTATKAYNVTQAFGR\n",
      "                     RGPEQTQGNFGDQELIRQGTDYKHWPQIAQFAPSASAFFGMSRIGMEVTPSGTWLTYT\n",
      "                     GAIKLDDKDPNFKDQVILLNKHIDAYKTFPPTEPKKDKKKKADETQALPQRQKKQQTV\n",
      "                     TLLPAADLDDFSKQLQQSMSSADSTQA\"\n",
      "     gene            29521..29637\n",
      "                     /gene=\"ORF10\"\n",
      "     CDS             29521..29637\n",
      "                     /gene=\"ORF10\"\n",
      "                     /codon_start=1\n",
      "                     /product=\"ORF10 protein\"\n",
      "                     /protein_id=\"QSB46707.1\"\n",
      "                     /translation=\"MGYINVFAFPFTIYSLLLCRMNSRNYIAQVDVVNFNLT\"\n",
      "     stem_loop       29572..29607\n",
      "                     /gene=\"ORF10\"\n",
      "                     /note=\"Coronavirus 3' UTR pseudoknot stem-loop 1\"\n",
      "     stem_loop       29592..29620\n",
      "                     /gene=\"ORF10\"\n",
      "                     /note=\"Coronavirus 3' UTR pseudoknot stem-loop 2\"\n",
      "     stem_loop       29691..29731\n",
      "                     /note=\"Coronavirus 3' stem-loop II-like motif (s2m)\"\n",
      "ORIGIN      \n",
      "        1 aactttcgat ctcttgtaga tctgttctct aaacgaactt taaaatctgt gtggctgtca\n",
      "       61 ctcggctgca tgcttagtgc actcacgcag tataattaat aactaattac tgtcgttgac\n",
      "      121 aggacacgag taactcgtct atcttctgca ggctgcttac ggtttcgtcc gtgttgcagc\n",
      "      181 cgatcatcag cacatctagg ttttgtccgg gtgtgaccga aaggtaagat ggagagcctt\n",
      "      241 gtccctggtt tcaacgagaa aacacacgtc caactcagtt tgcctgtttt acaggttcgc\n",
      "      301 gacgtgctcg tacgtggctt tggagactcc gtggaggagg tcttatcaga ggcacgtcaa\n",
      "      361 catcttaaag atggcacttg tggcttagta gaagttgaaa aaggcgtttt gcctcaactt\n",
      "      421 gaacagccct atgtgttcat caaacgttcg gatgctcgaa ctgcacctca tggtcatgtt\n",
      "      481 atggttgagc tggtagcaga actcgaaggc attcagtacg gtcgtagtgg tgagacactt\n",
      "      541 ggtgtccttg tccctcatgt gggcgaaata ccagtggctt accgcaaggt tcttcttcgt\n",
      "      601 aagaacggta ataaaggagc tggtggccat agttacggcg ccgatctaaa gtcatttgac\n",
      "      661 ttaggcgacg agcttggcac tgatccttat gaagattttc aagaaaactg gaacactaaa\n",
      "      721 catagcagtg gtgttacccg tgaactcatg cgtgagctta acggaggggc atacactcgc\n",
      "      781 tatgtcgata acaacttctg tggccctgat ggctaccctc ttgagtgcat taaagacctt\n",
      "      841 ctagcacgtg ctggtaaagc ttcatgcact ttgtccgaac aactggactt tattgacact\n",
      "      901 aagaggggtg tatactgctg ccgtgaacat gagcatgaaa ttgcttggta cacggaacgt\n",
      "      961 tctgaaaaga gctatgaatt gcagacacct tttgaaatta aattggcaaa gaaatttgac\n",
      "     1021 atcttcaatg gggaatgtcc aaattttgta tttcccttaa attccataat caagactatt\n",
      "     1081 caaccaaggg ttgaaaagaa aaagcttgat ggctttatgg gtagaattcg atctgtctat\n",
      "     1141 ccagttgcgt caccaaatga atgcaaccaa atgtgccttt caactctcat gaagtgtgat\n",
      "     1201 cattgtggtg aaacttcatg gcagacgggc gattttgtta aagccacttg cgaattttgt\n",
      "     1261 ggcactgaga atttgactaa agaaggtgcc actacttgtg gttacttacc ccaaaatgct\n",
      "     1321 gttgttaaaa tttattgtcc agcatgtcac aattcagaag taggacctga gcatagtctt\n",
      "     1381 gccgaatacc ataatgaatc tggcttgaaa accattcttc gtaagggtgg tcgcactatt\n",
      "     1441 gcctttggag gctgtgtgtt ctcttatgtt ggttgccata acaagtgtgc ctattgggtt\n",
      "     1501 ccacgtgcta gcgctaacat aggttgtaac catacaggtg ttgttggaga aggttccgaa\n",
      "     1561 ggtcttaatg acaaccttct tgaaatactc caaaaagaga aagtcaacat caatattgtt\n",
      "     1621 ggtgacttta aacttaatga agagatcgcc attattttgg catctttttc tgcttccaca\n",
      "     1681 agtgcttttg tggaaactgt gaaaggtttg gattataaag cattcaaaca aattgttgaa\n",
      "     1741 tcctgtggta attttaaagt tacaaaagga aaagctaaaa aaggtgcctg gaatattggt\n",
      "     1801 gaacagaaat caatactgag tcctctttat gcatttgcat cagaggctgc tcgtgttgta\n",
      "     1861 cgatcaattt tctcccgcac tcttgaaact gctcaaaatt ctgtgcgtgt tttacagaag\n",
      "     1921 gccgctataa caatactaga tggaatttca cagtattcac tgagactcat tgatgctatg\n",
      "     1981 atgttcacat ctgatttggc tactaacaat ctagttgtaa tggcctacat tacaggtggt\n",
      "     2041 gttgttcagt tgacttcgca gtggctaact aacatctttg gcactgttta tgaaaaactc\n",
      "     2101 aaacccgtcc ttgattggct tgaagagaag tttaaggaag gtgtagagtt tcttagagac\n",
      "     2161 ggttgggaaa ttgttaaatt tatctcaacc tgtgcttgtg aaattgtcgg tggacaaatt\n",
      "     2221 gtcacctgtg caaaggaaat taaggagagt gttcagacat tctttaagct tgtaaataaa\n",
      "     2281 tttttggctt tgtgtgctga ctctatcatt attggtggag ctaaacttaa agccttgaat\n",
      "     2341 ttaggtgaaa catttgtcac gcactcaaag ggattgtaca gaaagtgtgt taaatccaga\n",
      "     2401 gaagaaactg gcctactcat gcctctaaaa gccccaaaag aaattatctt cttagaggga\n",
      "     2461 gaaacacttc ccacagaagt gttaacagag gaagttgtct tgaaaactgg tgatttacaa\n",
      "     2521 ccattagaac aacctactag tgaagctgtt gaagctccat tggttggtac accagtttgt\n",
      "     2581 attaacgggc ttatgttgct cgaaatcaaa gacacagaaa agtactgtgc ccttgcacct\n",
      "     2641 aatatgatgg taacaaacaa taccttcaca ctcaaaggcg gtgcaccaac aaaggttact\n",
      "     2701 tttggtgatg acactgtgat agaantgcaa ggttacaaga gtgtgaatat cacttttgaa\n",
      "     2761 cttgatgaaa ggattgataa agtacttaat gagaagtgct ctgcctatac agttgaactc\n",
      "     2821 ggtacagaag taaatgagtt cgcctgtgtt gtggcagatg ctgtcataaa aactttgcaa\n",
      "     2881 ccagtatctg aattacttac accactgggc attgatttag atgagtggag tatggctaca\n",
      "     2941 tactacttat ttgatgagtc tggtgagttt aaattggctt cacatatgta ttgttctttt\n",
      "     3001 taccctccag atgaggatga agaagaaggt gattgtgaag aagaagagtt tgagccatca\n",
      "     3061 actcaatatg agtatggtac tgaagatgat taccaaggta aacctttgga atttggtgcc\n",
      "     3121 acttctgctg ctcttcaacc tgaagaagag caagaagaag attggttaga tgatgatagt\n",
      "     3181 caacaaactg ttggtcaaca agacggcagt gaggacaatc agacaactac tattcaaaca\n",
      "     3241 attgttgagg ttcaacctca attagagatg gaacttacac cagttgttca gactattgaa\n",
      "     3301 gtgaatagtt ttagtggtta tttaaaactt actgacaatg tatacattaa aaatgcagac\n",
      "     3361 attgtggaag aagctaaaaa ggtaaaacca acagtggttg ttaatgcagc caatgtttac\n",
      "     3421 cttaaacatg gaggaggtgt tgcaggagcc ttaaataagg ctactaacaa tgccatgcaa\n",
      "     3481 gttgaatctg atgattacat agctactaat ggaccactta aagtgggtgg tagttgtgtt\n",
      "     3541 ttaagcggac acaatcttgc taaacactgt cttcatgttg tcggcccaaa tgttaacaaa\n",
      "     3601 ggtgaagaca ttcaacttct taagagtgct tatgaaaatt ttaatcagca cgaagttcta\n",
      "     3661 cttgcaccat tattatcagc tggtattttt ggtgctgacc ctatacattc tttaagagtt\n",
      "     3721 tgtgtagata ctgttcgcac aaatgtctac ttagctgtct ttgataaaaa tctctatgac\n",
      "     3781 aaacttgttt caagcttttt ggaaatgaag agtgaaaagc aagttgaaca aaagatcgct\n",
      "     3841 gagattccta aagaggaagt taagccattt ataactgaaa gtaaaccttc agttgaacag\n",
      "     3901 agaaaacaag atgataagaa aatcaaagct tgtgttgaag aagttacaac aactctggaa\n",
      "     3961 gaaactaagt tcctcacaga aaacttgtta ctttatattg acattaatgg caatcttcat\n",
      "     4021 ccagattctg ccactcttgt tagtgacatt gacatcactt tcttaaagaa agatgctcca\n",
      "     4081 tatatagtgg gtgatgttgt tcaagagggt gttttaactg ctgtggttat acctactaaa\n",
      "     4141 aaggctggtg gcactactga aatgctagcg aaagctttga gaaaagtgcc aacagacaat\n",
      "     4201 tatataacca cttacccggg tcagggttta aatggttaca ctgtagagga ggcaaagaca\n",
      "     4261 gtgcttaaaa agtgtaaaag tgccttttac attctaccat ctattatctc taatgagaag\n",
      "     4321 caagaaattc ttggaactgt ttcttggaat ttgcgagaaa tgcttgcaca tgcagaagaa\n",
      "     4381 acacgcaaat taatgcctgt ctgtgtggaa actaaagcca tagtttcaac tatacagcgt\n",
      "     4441 aaatataagg gtattaaaat acaagagggt gtggttgatt atggtgctag attttacttt\n",
      "     4501 tacaccagta aaacaactgt agcgtcactt atcaacacac ttaacgatct aaatgaaact\n",
      "     4561 cttgttacaa tgccacttgg ctatgtaaca catggcttaa atttggaaga agctgctcgg\n",
      "     4621 tatatgagat ctctcaaagt gccagctaca gtttctgttt cttcacctga tgctgttaca\n",
      "     4681 gcgtataatg gttatcttac ttcttcttct aaaacacctg aggaacattt tattgaaacc\n",
      "     4741 atctcacttg ctggttccta taaagattgg tcctattctg gacaatctac acaactaggt\n",
      "     4801 atagaatttc ttaagagagg tgataaaagt gtatattaca ctagtaatcc taccacattc\n",
      "     4861 cacctagatg gtgaagttat cacctttgac aatcttaaga cacttctttc tttgagagaa\n",
      "     4921 gtgaggacta ttaaggtgtt tacaacagta gacaacatta acctccacac gcaagttgtg\n",
      "     4981 gacatgtcaa tgacatatgg acaacagttt ggtccaactt atttggatgg agctgatgtt\n",
      "     5041 actaaaataa aacctcataa ttcacatgaa ggtaaaacat tttatgtttt acctaatgat\n",
      "     5101 gacactctac gtgttgaggc ttttgagtac taccacacaa ctgatcctag ttttctgggt\n",
      "     5161 aggtacatgt cagcattaaa tcacactaaa aagtggaaat acccacaagt taatggttta\n",
      "     5221 acttctatta aatgggcaga taacaactgt tatcttgcca ctgcattgtt aacactccaa\n",
      "     5281 caaatagagt tgaagtttaa tccacctgct ctacaagatg cttattacag agcaagggct\n",
      "     5341 ggtgaagctg ctaacttttg tgcacttatc ttagcctact gtaataagac agtaggtgag\n",
      "     5401 ttaggtgatg ttagagaaac aatgagttac ttgtttcaac atgccaattt agattcttgc\n",
      "     5461 aaaagagtct tgaatgtggt gtgtaaaact tgtggacaac agcagacaac ccttaagggt\n",
      "     5521 gtagaagctg ttatgtacat gggcacactt tcttatgaac aatttaagaa aggtgttcag\n",
      "     5581 ataccttgta cgtgtggtaa acaagctaca aaatatctag tacaacagga gtcacctttt\n",
      "     5641 gttatgatgt cagcaccacc tgctcagtat gaacttaagc atggtacatt tacttgtgct\n",
      "     5701 agtgagtaca ctggtaatta ccagtgtggt cactataaac atataacttc taaagaaact\n",
      "     5761 ttgtattgca tagacggtgc tttacttaca aagtcctcag aatacaaagg tcctattacg\n",
      "     5821 gatgttttct acaaagaaaa cagttacaca acaaccataa aaccagttac ttataaattg\n",
      "     5881 gatggtgttg tttgtacaga aattgaccct aagttggaca attattataa gaaagacaat\n",
      "     5941 tcttatttca cagagcaacc aattgatctt gtaccaaacc aaccatatcc aaacgcaagc\n",
      "     6001 ttcgataatt ttaagtttgt atgtgataat atcaaatttg ctgatgattt aaaccagtta\n",
      "     6061 actggttata agaaacctgc ttcaagagag cttaaagtta catttttccc tgacttaaat\n",
      "     6121 ggtgatgtgg tggctattga ttataaacac tacacaccct cttttaagaa aggagctaaa\n",
      "     6181 ttgttacata aacctattgt ttggcatgtt aacaatgcaa ctaataaagc cacgtataaa\n",
      "     6241 ccaaatacct ggtgtatacg ttgtctttgg agcacaaaac cagttgaaac atcaaattcg\n",
      "     6301 tttgatgtac tgaagtcaga ggacgcgcag ggaatggata atcttgcctg cgaagatcta\n",
      "     6361 aaaccagtct ctgaagaagt agtggaaaat cctaccatac agaaagacgt tcttgagtgt\n",
      "     6421 aatgtgaaaa ctaccgaagt tgtaggagac attatactta aaccagcaaa taatagttta\n",
      "     6481 aaaattatag aagaggttgg ccacacagat ctaatggctg cttatgtaga caattctagt\n",
      "     6541 cttactatta agaaacctaa tgaattatct agagtattag gtttgaaaac ccttgctact\n",
      "     6601 catggtttag ctgctgttaa tagtgtccct tgggatacta tagctaatta tgctaagcct\n",
      "     6661 tttcttaaca aagttgttag tacaactact aacatagtta cacggtgttt aaaccgtgtt\n",
      "     6721 tgtactaatt atatgcctta tttctttact ttattgctac aattgtgtac ttttactaga\n",
      "     6781 agtacaaatt ctagaattaa agcatctatg ccgactacta tagcaaanaa tactgttaag\n",
      "     6841 agtgtcggta aattttgtct agaggcttca tttaattatt tgaagtcacc taatttttct\n",
      "     6901 aaactgataa atattataat ttggttttta ctattaagtg tttgcctagg ttctttaatc\n",
      "     6961 tactcaaccg ctgctttagg tgttttaatg tctaatttag gcatgccttc ttactgtact\n",
      "     7021 ggttacagag aannctattt gaactctact aatgtcacta ttgcaaccta ctgtactggt\n",
      "     7081 tctatacctt gtagtgtttg tcttagtggt ttagattctt tagacaccta tccttcttta\n",
      "     7141 gaaactatac aaattaccat ttcatctttt aaatgggatt taactgcttt tggcttagtt\n",
      "     7201 gcagagtggt ttttggcata tattcttttc acnaggtttt tctatgtact tggattggct\n",
      "     7261 gcaatcatgc aattgttttt cagctatttt gcagtacatt ttattagtaa ttcttggctt\n",
      "     7321 atgtggttaa taattaatct tgtacaaatg gccccgattt cagctatggt tagaatgtac\n",
      "     7381 atcttctttg catcatttta ttatgtatgg aaaagttatg tgcatgttgt agacggttgt\n",
      "     7441 aattcatcaa cttgtatgat gtgttacaaa cgtaatagag caacaagagt cgaatgtaca\n",
      "     7501 actattgtta atggtgttag aaggtccttt tatgtctatg ctaatggagg taaaggcttt\n",
      "     7561 tgcaaactac acaattggaa ttgtgttaat tgtgatacat tctgtgctgg tagtacattt\n",
      "     7621 attagtgatg aagttgcgag agacttgtca ctacagttta aaagaccaat aaatcctact\n",
      "     7681 gaccagtctt cttacatcgt tgatagtgtt acagtgaaga atggttccat ccatctttac\n",
      "     7741 tttgataaag ctggtcaaaa gacttatgaa agacattctc tctctcattt tgttaactta\n",
      "     7801 gacaacctga gagctaataa cactaaaggt tcattgccta ttaatgttat agtttttgat\n",
      "     7861 ggtaaatcaa aatgtgaaga atcatctgca aaatcagcgt ctgtttacta cagtcagctt\n",
      "     7921 atgtgtcaac ctatactgtt actagatcag gcattagtgt ctgatgttgg tgatagtgcg\n",
      "     7981 gaagttgcag ttaaaatgtt tgatgcttac gttaatacgt tttcatcaac ttttaacgta\n",
      "     8041 ccaatagaaa aactcaaaac actagttgca actgcagaag ctgaacttgc aaagaatgtg\n",
      "     8101 tccttagaca atgtcttatc tacttttatt tcagcagctc ggcaagggtt tgttgattca\n",
      "     8161 gatgtagaaa ctaaagatgt tgttgaatgt cttaaattgt cacatcaatc tgacatagaa\n",
      "     8221 gttactggcg atagttgtaa taactatatg ctcacctata acaaagttga aaacatgaca\n",
      "     8281 ccccgtgacc ttggtgcttg tattgactgt agtgcgcgtc atattaatgc gcaggtagca\n",
      "     8341 aaaagtcaca acattgcttt gatatggaac gttaaagatt tcatgtcatt gtctgaacaa\n",
      "     8401 ctacgaaaac aaatacgtag tgctgctaaa aagaataact taccttttaa gttgacatgt\n",
      "     8461 gcaactacta gacaagttgt taatgttgta acaacaaaga tagcacttaa gggtggtaaa\n",
      "     8521 attgttaata attggttgaa gcagttaatt aaagttacac ttgtgttcct ttttgttgct\n",
      "     8581 gctattttct atttaataac acctgttcat gtcatgtcta aacatactga cttttcaagt\n",
      "     8641 gaaatcatag gatacaaggc tattgatggt ggtgtcactc gtgacatagc atctacagat\n",
      "     8701 acttgttttg ctaacaaaca tgctgatttt gacacatggt ttagccagcg tggtggtagt\n",
      "     8761 tatactaatg acaaagcttg cccattgatt gctgcagtca taacaagaga agtgggtttt\n",
      "     8821 gtcgtgcctg gtttgcctgg cacgatatta cgcacaacta atggtgactt tttgcatttc\n",
      "     8881 ttacctagag tttttagtgc agttggtaac atctgttaca caccatcaaa acttatagag\n",
      "     8941 tacactgact ttgcaacatc agcttgtgtt ttggctgctg aatgtacaat ttttaaagat\n",
      "     9001 gcttctggta agccagtacc atattgttat gataccaatg tactagaagg ttctgttgct\n",
      "     9061 tatgaaagtt tacgccctga cacacgttat gtgctcatgg atggctctat tattcaattt\n",
      "     9121 cctaacacct accttgaagg ttctgttaga gtggtaacaa cttttgattc tgagtactgt\n",
      "     9181 aggcacggca cttgtgaaag atcagaagct ggtgtttgtg tatctactag tggtagatgg\n",
      "     9241 gtacttaaca atgattatta cagatcttta ccaggagttt tctgtggtgt agatgctgta\n",
      "     9301 aatttactta ctaatatgtt tacaccacta attcaaccta ttggtgcttt ggacatatca\n",
      "     9361 gcatctatag tagctggtgg tattgtagct atcgtagtaa catgccttgc ctactatttt\n",
      "     9421 atgaggttta gaagagcttt tggtgaatac agtcatgtag ttgcctttaa tactttacta\n",
      "     9481 ttccttatgt cattcactgt actctgttta acaccagttt actcattctt acctggtgtt\n",
      "     9541 tattctgtta tttacttgta cttgacattt tatcttacta atgatgtttc ttttttagca\n",
      "     9601 catattcagt ggatggttat gttcacacct ttagtacctt tctggataac aattgcttat\n",
      "     9661 atcatttgta tttccacaaa gcatttctat tggttcttta gtaattacct aaagagacgt\n",
      "     9721 gtagtcttta atggtgtttc ctttagtact tttgaagaag ctgcgctgtg cacctttttg\n",
      "     9781 ttaaataaag aaatgtatct aaagttgcgt agtgatgtgc tattacctct tacgcaatat\n",
      "     9841 aatagatact tagctcttta taataagtac aagtatttta gtggagcaat ggatacaact\n",
      "     9901 agctacagag aagctgcttg ttgtcatctc gcaaaggctc tcaatgactt cagtaactca\n",
      "     9961 ggttctgatg ttctttacca accaccacaa acctctatca cctcagctgt tttgcagagt\n",
      "    10021 ggttttagaa aaatggcatt cccatctggt aaagttgagg gttgtatggt acaagtaact\n",
      "    10081 tgtggtacaa ctacacttaa cggtctttgg cttgatgacg tagtttactg tccaagacat\n",
      "    10141 gtgatctgca cctctgaaga catgcttaac cctaattatg aagatttact cattcgtaag\n",
      "    10201 tctaatcata atttcttggt acaggctggt aatgttcaac tcagggttat tggacattct\n",
      "    10261 atgcaaaatt gtgtacttaa gtttaaggtt gatacagcca atcctaagac acctaagtat\n",
      "    10321 aagtttgttc gcattcaacc aggacagact ttttcagtgt tagcttgtta caatggttca\n",
      "    10381 ccatctggtg tttaccaatg tgctatgagg cccaatttca ctattaaggg ttcattcctt\n",
      "    10441 aatggttcat gtggtagtgt tggttttaac atagattatg actgtgtctc tttttgttac\n",
      "    10501 atgcaccata tggaattacc aactggagtt catgctggca cagacttaga aggtaacttt\n",
      "    10561 tatggacctt ttgttgacag gcaaacagca caagcagctg gtacggacac aactattaca\n",
      "    10621 gttaatgttt tagcttggtt gtacgctgct gttataaatg gagacaggtg gtttctcaat\n",
      "    10681 cgatttacca caactcttaa tgactttaac cttgtggcta tgaagtacaa ttatgaacct\n",
      "    10741 ctaacacaag accatgttga catactagga cctctttctg ctcaaactgg aattgccgtt\n",
      "    10801 ttagatatgt gtgcttcatt aaaagaatta ctgcaaaatg gtatgaatgg acgtaccata\n",
      "    10861 ttgggtagtg ctttattaga agatgaattt acaccttttg atgttgttag acaatgctca\n",
      "    10921 ggtgttactt tccaaagtgc agtgaaaaga acaatcaagg gtacacacca ctggttgtta\n",
      "    10981 ctcacaattt tgacttcact tttagtttta gtccagagta ctcaatggtc tttgttcttt\n",
      "    11041 tttttgtatg aaaatgcctt tttacctttt gctatgggta ttattgctat gtctgctttt\n",
      "    11101 gcaatgatgt ttgtcaaaca taagcatgca tttctctgtt tgtttttgtt accttctctt\n",
      "    11161 gccactgtag cttattttaa tatggtctat atgcctgcta gttgggtgat gcgtattatg\n",
      "    11221 acatggttgg atatggttga tactagtttg tctggtttta agctaaaaga ctgtgttatg\n",
      "    11281 tatgcatcag ctgtagtgtt actaatcctt atgacagcaa gaactgtgta tgatgatggt\n",
      "    11341 gctaggagag tgtggacact tatgaatgtc ttgacactcg tttataaagt ttattatggt\n",
      "    11401 aatgctttag atcaagccat ttccatgtgg gctcttataa tctctgttac ttctaactac\n",
      "    11461 tcaggtgtag ttacaactgt catgtttttg gccagaggta ttgtttttat gtgtgttgag\n",
      "    11521 tattgcccta ttttcttcat aactggtaat acacttcagt gtataatgct agtttattgt\n",
      "    11581 ttcttaggct atttttgtac ttgttacttt ggcctctttt gtttactcaa ccgctacttt\n",
      "    11641 agactgactc ttggtgttta tgattactta gtttctacac aggagtttag atatatgaat\n",
      "    11701 tcacagggac tactcccacc caagaatagc atagatgcct tcaaactcaa cattaaattg\n",
      "    11761 ttgggtgttg gtggcaaacc ttgtatcaaa gtagccactg tacagtctaa aatgtcagat\n",
      "    11821 gtaaagtgca catcagtagt cttactctca gttttgcaac aactcagagt agaatcatca\n",
      "    11881 tctaaattgt gggctcaatg tgtccagtta cacaatgaca ttctcttagc taaagatact\n",
      "    11941 actgaagcct ttgaaaaaat ggtttcacta ctttctgttt tgctttccat gcagggtgct\n",
      "    12001 gtagacataa acaagctttg tgaagaaatg ctggacaaca gggcaacctt acaagctata\n",
      "    12061 gcctcagagt ttagttccct tccatcatat gcagcttttg ctactgctca agaagcttat\n",
      "    12121 gagcaggctg ttgctaatgg tgattctgaa gttgttctta aaaagttgaa gaagtctttg\n",
      "    12181 aatgtggcta aatctgaatt tgaccgtgat gcagccatgc aacgtaagtt ggaaaagatg\n",
      "    12241 gctgatcaag ctatgaccca aatgtataaa caggctagat ctgaggacaa gagggcaaaa\n",
      "    12301 gttactagtg ctatgcagac aatgcttttc actatgctta gaaagttgga taatgatgca\n",
      "    12361 ctcaacaaca ttatcaacaa tgcaagagat ggttgtgttc ccttgaacat aatacctctt\n",
      "    12421 acaacagcag ccaaactaat ggttgtcata ccagactata acacatataa aaatacgtgt\n",
      "    12481 gatggtacaa catttactta tgcatcagca ttgtgggaaa tccaacaggt tgtagatgca\n",
      "    12541 gatagtaaaa ttgttcaact tagtgaaatt agtatggaca attcacctaa tttagcatgg\n",
      "    12601 cctcttattg taacagcttt aagggccaat tctgctgtca aattacagaa taatgagctt\n",
      "    12661 agtcctgttg cactacgaca gatgtcttgt gctgccggta ctacacaaac tgcttgcact\n",
      "    12721 gatgacaatg cgttagctta ctacaacaca acaaagggag gtaggtttgt acttgcactg\n",
      "    12781 ttatccgatt tacaggattt gaaatgggct agattcccta agagtgatgg aactggtact\n",
      "    12841 atctatacag aactggaacc accttgtagg tttgttacag acacacctaa aggtcctaaa\n",
      "    12901 gtgaagtatt tatactttat taaaggatta aacaacctaa atagaggtat ggtacttggt\n",
      "    12961 agtttagctg ccacagtacg tctacaagct ggtaatgcaa cagaagtgcc tgccaattca\n",
      "    13021 actgtattat ctttctgtgc ttttgctgta gatgctgcta aagcttacaa agattatcta\n",
      "    13081 gctagtgggg gacaaccaat cactaattgt gttaagatgt tgtgtacaca cactggtact\n",
      "    13141 ggtcaggcaa taacagttac accggaagcc aatatggatc aagaatcctt tggtggtgca\n",
      "    13201 tcgtgttgtc tgtactgccg ttgccacata gatcatccaa atcctaaagg attttgtgac\n",
      "    13261 ttaaaaggta agtatgtaca aatacctaca acttgtgcta atgaccctgt gggttttaca\n",
      "    13321 cttaaaaaca cagtctgtac cgtctgcggt atgtggaaag gttatggctg tagttgtgat\n",
      "    13381 caactccgcg aacccatgct tcagtcagct gatgcacaat cgtttttaaa cgggtttgcg\n",
      "    13441 gtgtaagtgc agcccgtctt acaccgtgcg gcacaggcac tagtactgat gtcgtataca\n",
      "    13501 gggcttttga catctacaat gataaagtag ctggttttgc taaattccta aaaactaatt\n",
      "    13561 gttgtcgctt ccaagaaaag gacgaagatg acaatttaat tgattcttac tttgtagtta\n",
      "    13621 agagacacac tttctctaac taccaacatg aagaaacaat ttataattta cttaaggatt\n",
      "    13681 gtccagctgt tgctaaacat gacttcttta agtttagaat agacggtgac atggtaccac\n",
      "    13741 atatatcacg tcaacgtctt actaaataca caatggcaga cctcgtctat gctttaaggc\n",
      "    13801 attttgatga aggtaattgt gacacattaa aagaaatact tgtcacatac aattgttgtg\n",
      "    13861 atgatgatta tttcaataaa aaggactggt atgattttgt agaaaaccca gatatattac\n",
      "    13921 gcgtatacgc caacttaggt gaacgtgtac gccaagcttt gttaaaaaca gtacaattct\n",
      "    13981 gtgatgccat gcgaaatgct ggtattgttg gtgtactgac attagataat caagatctca\n",
      "    14041 atggtaactg gtatgatttc ggtgatttca tacaaaccac gccaggtagt ggagttcctg\n",
      "    14101 ttgtagattc ttattattca ttgttaatgc ctatattaac cttgaccagg gctttaactg\n",
      "    14161 cagagtcaca tgttgacact gacttaacaa agccttacat taagtgggat ttgttaaaat\n",
      "    14221 atgacttcac ggaagagagg ttaaaactct ttgaccgtta ttttaaatat tgggatcaga\n",
      "    14281 cataccaccc aaattgtgtt aactgtttgg atgacagatg cattctgcat tgtgcaaact\n",
      "    14341 ttaatgtttt attctctaca gtgttcccac ttacaagttt tggaccacta gtgagaaaaa\n",
      "    14401 tatttgttga tggtgttcca tttgtagttt caactggata ccacttcaga gagctaggtg\n",
      "    14461 ttgtacataa tcaggatgta aacttacata gctctagact tagttttaag gaattacttg\n",
      "    14521 tgtatgctgc tgaccctgct atgcacgctg cttctggtaa tctattacta gataaacgca\n",
      "    14581 ctacgtgctt ttcagtagct gcacttacta acaatgttgc ttttcaaact gtcaaacccg\n",
      "    14641 gtaattttaa caaagacttc tatgactttg ctgtgtctaa gggtttcttt aaggaaggaa\n",
      "    14701 gttctgttga attaaaacac ttcttctttg ctcaggatgg taatgctgct atcagcgatt\n",
      "    14761 atgactatta tcgttataat ctaccaacaa tgtgtgatat cagacaacta ctatttgtag\n",
      "    14821 ttgaagttgt tgataagtac tttgattgtt acgatggtgg ctgtattaat gctaaccaag\n",
      "    14881 tcatcgtcaa caacctagac aaatcagctg gttttccatt taataaatgg ggtaaggcta\n",
      "    14941 gactttatta tgattcaatg agttatgagg atcaagatgc acttttcgca tatacaaaac\n",
      "    15001 gtaatgtcat ccctactata actcaaatga atcttaagta tgccattagt gcaaagaata\n",
      "    15061 gagctcgcac cgtagctggt gtctctatct gtagtactat gaccaataga cagtttcatc\n",
      "    15121 aaaaattatt gaaatcaata gccgccacta gaggagctac tgtagtaatt ggaacaagca\n",
      "    15181 aattctatgg tggttggcac aacatgttaa aaactgttta tagtgatgta gaaaaccctc\n",
      "    15241 accttatggg ttgggattat cctaaatgtg atagagccat gcctaacatg cttagaatta\n",
      "    15301 tggcctcact tgttcttgct cgcaaacata caacgtgttg tagcttgtca caccgtttct\n",
      "    15361 atagattagc taatgagtgt gctcaagtat tgagtgaaat ggtcatgtgt ggcggttcac\n",
      "    15421 tatatgttaa accaggtgga acctcatcag gagatgccac aactgcttat gctaatagtg\n",
      "    15481 tttttaacat ttgtcaagct gtcacggcca atgttaatgc acttttatct actgatggta\n",
      "    15541 acaaaattgc cgataagtat gtccgcaatt tacaacacag actttatgag tgtctctata\n",
      "    15601 gaaatagaga tgttgacaca gactttgtga atgagtttta cgcatatttg cgtaaacatt\n",
      "    15661 tctcaatgat gatactctct gacgatgctg ttgtgtgttt caatagcact tatgcatctc\n",
      "    15721 aaggtctagt ggctagcata aagaacttta agtcagttct ttattatcaa aacaatgttt\n",
      "    15781 ttatgtctga agcaaaatgt tggactgaga ctgaccttac taaaggacct catgaatttt\n",
      "    15841 gctctcaaca tacaatgcta gttaaacagg gtgatgatta tgtgtacctt ccttacccag\n",
      "    15901 atccatcaag aatcctaggg gccggctgtt ttgtagatga tatcgtaaaa acagatggta\n",
      "    15961 cacttatgat tgaacggttc gtgtctttag ctatagatgc ttacccactt actaaacatc\n",
      "    16021 ctaatcagga gtatgctgat gtctttcatt tgtacttaca atacataaga aagctacatg\n",
      "    16081 atgagttaac aggacacatg ttagacatgt attctgttat gcttactaat gataacactt\n",
      "    16141 caaggtattg ggaacctgag ttttatgagg ctatgtacac accgcataca gtcttacagg\n",
      "    16201 ctgttggggc ttgtgttctt tgcaattcac agacttcatt aagatgtggt gcttgcatac\n",
      "    16261 gtagaccatt cttatgttgt aaatgctgtt acgaccatgt catatcaaca tcacataaat\n",
      "    16321 tagtcttgtc tgttaatccg tatgtttgca atgctccagg ttgtgatgtc acagatgtga\n",
      "    16381 ctcaacttta cttaggaggt atgagctatt attgtaaatc acataaacca cccattagtt\n",
      "    16441 ttccattgtg tgctaatgga caagtttttg gtttatataa aaatacatgt gttggtagcg\n",
      "    16501 ataatgttac tgactttaat gcaattgcaa catgtgactg gacaaatgct ggtgattaca\n",
      "    16561 ttttagctaa cacctgtact gaaagactca agctttttgc agcagaaacg ctcaaagcta\n",
      "    16621 ctgaggagac atttaaactg tcttatggta ttgctactgt acgtgaagtg ctgtctgaca\n",
      "    16681 gagaattaca tctttcatgg gaagttggta aacctagacc accacttaac cgaaattatg\n",
      "    16741 tctttactgg ttatcgtgta actaaaaaca gtaaagtaca aataggagag tacacctttg\n",
      "    16801 aaaaaggtga ctatggtgat gctgttgttt accgaggtac aacaacttac aaattaaatg\n",
      "    16861 ttggtgatta ttttgtgctg acatcacata cagtaatgcc attaagtgca cctacactag\n",
      "    16921 tgccacaaga gcactatgtt agaattactg gcttataccc aacactcaat atctcagatg\n",
      "    16981 agttttctag caatgttgca aattatcaaa aggttggtat gcaaaagtat tctacactcc\n",
      "    17041 agggaccacc tggtactggt aagagtcatt ttgctattgg cctagctctc tactaccctt\n",
      "    17101 ctgctcgcat agtgtataca gcttgctctc atgccgctgt tgatgcacta tgtgagaagg\n",
      "    17161 cattaaaata tttgcctata gataaatgta gtagaattat acctgcacgt gctcgtgtag\n",
      "    17221 agtgttttga taaattcaaa gtgaattcaa cattagaaca gtatgtcttt tgtactgtaa\n",
      "    17281 atgcattgcc tgagacgaca gcagatatag ttgtctttga tgaaatttca atggccacaa\n",
      "    17341 attatgattt gagtgttgtc aatgccagat tacgtgctaa gcactatgtg tacattggcg\n",
      "    17401 accctgctca attacctgca ccacgcacat tgctaactaa gggcacacta gaaccagaat\n",
      "    17461 atttcaattc agtgtgtaga cttatgaaaa ctataggtcc agacatgttc ctcggaactt\n",
      "    17521 gtcggcgttg tcctgctgaa attgttgaca ctgtgagtgc tttggtttat gataataagc\n",
      "    17581 ttaaagcaca taaagacaaa tcagctcaat gctttaaaat gttttataag ggtgttatca\n",
      "    17641 cgcatgatgt ttcatctgca attaacaggc cacaaatagg cgtggtaaga gaattcctta\n",
      "    17701 cacgtaaccc tgcttggaga aaagctgtct ttatttcacc ttataattca cagaatgctg\n",
      "    17761 tagcctcaaa gattttggga ctaccaactc aaactgttga ttcatcacag ggctcagaat\n",
      "    17821 atgactatgt catattcact caaaccactg aaacagctca ctcttgtaat gtaaacagat\n",
      "    17881 ttaatgttgc tattaccaga gcaaaagtag gcatactttg cataatgtct gatagagacc\n",
      "    17941 tttatgacaa gttgcaattt acaagtcttg aaattccacg taggaatgtg gcaactttac\n",
      "    18001 aagctgaaaa tgtaacagga ctctttaaag attgtagtaa ggtaatcact gggttacatc\n",
      "    18061 ctacacaggc acctacacac ctcagtgttg acactaaatt caaaactgaa ggtttatgtg\n",
      "    18121 ttgacatacc tggcatacct aaggacatga cctatagaag actcatctct atgatgggtt\n",
      "    18181 ttaaaatgaa ttatcaagtt aatggttacc ctaacatgtt tatcacccgc gaagaagcta\n",
      "    18241 taagacatgt acgtgcatgg attggcttcg atgtcgaggg gtgtcatgct actagagaag\n",
      "    18301 ctgttggtac caatttacct ttacagctag gtttttctac aggtgttaac ctagttgctg\n",
      "    18361 tacctacagg ttatgttgat acacctgata atacagattt ttccagagtt agtgctaaac\n",
      "    18421 caccgcctgg agatcaattt aaacacctca taccacttat gtacaaagga cttccttgga\n",
      "    18481 atgtagtgcg tataaagatt gtacaaatgt taagtgacac acttaaaaat ctctctgaca\n",
      "    18541 gagtcgtatt tgtcttatgg gcacatggct ttgagttgac atctatgaag tattttgtga\n",
      "    18601 aaataggacc tgagcgcacc tgttgtctat gtgatagacg tgccacatgc ttttccactg\n",
      "    18661 cttcagacac ttatgcctgt tggcatcatt ctattggatt tgattacgtc tataatccgt\n",
      "    18721 ttatgattga tgttcaacaa tggggtttta caggtaacct acaaagcaac catgatctgt\n",
      "    18781 attgtcaagt ccatggtaat gcacatgtag ctagttgtga tgcaatcatg actaggtgtc\n",
      "    18841 tagctgtcca cgagtgcttt gttaagcgtg ttgactggac tattgaatat cctataattg\n",
      "    18901 gtgatgaact gaagattaat gcggcttgta gaaaggttca acacatggtt gttaaagctg\n",
      "    18961 cattattagc agacaaattc ccagttcttc acgacattgg taaccctaaa gctattaagt\n",
      "    19021 gtgtacctca agctgatgta gaatggaagt tctatgatgc acagccttgt agtgacaaag\n",
      "    19081 cttataaaat agaagaatta ttctattctt atgccacaca ttctgacaaa ttcacagatg\n",
      "    19141 gtgtatgcct attttggaat tgcaatgtcg atagatatcc tgctaattcc attgtttgta\n",
      "    19201 gatttgacac tagagtgcta tctaacctta acttgcctgg ttgtgatggt ggcagnnnnn\n",
      "    19261 nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn\n",
      "    19321 nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn\n",
      "    19381 nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn\n",
      "    19441 nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn\n",
      "    19501 nnnnnnnnnn nnnnnncttt agcttgtggg tttacaaaca atttgatact tataacctct\n",
      "    19561 ggaacacttt tacaagactt cagagtttag aaaatgtggc ttttaatgtt gtaaataagg\n",
      "    19621 gacactttga tggacaacag ggtgaagtac cagtttctat cattaataac actgtttaca\n",
      "    19681 caaaagttga tggtgttgat gtagaattgt ttgaaaataa aacaacatta cctgttaatg\n",
      "    19741 tagcatttga gctttgggct aagcgcaaca ttaaaccagt accagaggtg aaaatactca\n",
      "    19801 ataatttggg tgtggacatt gctgctaata ctgtgatctg ggactacaaa agagatgctc\n",
      "    19861 cagcacatat atctactatt ggtgtttgtt ctatgactga catagccaag aaaccaactg\n",
      "    19921 aaacgatttg tgcaccactc actgtctttt ttgatggtag agttgatggt caagtagact\n",
      "    19981 tatttagaaa tgcccgtaat ggtgttctta ttacagaagg tagtgttaaa ggtttacaac\n",
      "    20041 catctgtagg tcccaaacaa gctagtctta atggagtcac attaattgga gaagccgtaa\n",
      "    20101 aaacacagtt caattattat aagaaagttg atggtgttgt ccaacaatta cctgaaactt\n",
      "    20161 actttactca gagtagaaat ttacaagaat ttaaacccag gagtcaaatg gaaattgatt\n",
      "    20221 tcttagaatt agctatggat gaattcattg aacggtataa attagaaggc tatgccttcg\n",
      "    20281 aacatatcgt ttatggagat tttagtcata gtcagttagg tggtttacat ctactgattg\n",
      "    20341 gactagctaa acgttttaag gaatcacctt ttgaattaga agattttatt cctatggaca\n",
      "    20401 gtacagttaa aaactatttc ataacagatg cgcaaacagg ttcatctaag tgtgtgtgtt\n",
      "    20461 ctgttattga tttattactt gatgattttg ttgaaataat aaaatcccaa gatttatctg\n",
      "    20521 tagtttctaa ggttgtcaaa gtgactattg actatacaga aatttcattt atgctttggt\n",
      "    20581 gtaaagatgg ccatgtagaa acattttacc caaaattaca atctagtcaa gcgtggcaac\n",
      "    20641 cgggtgttgc tatgcctaat ctttacaaaa tgcaaagaat gctattagaa aagtgtgacc\n",
      "    20701 ttcaaaatta tggtgatagt gcaacattac ctaaaggcat aatgatgaat gtcgcaaaat\n",
      "    20761 atactcaact gtgtcaatat ttaaacacat taacattagc tgtaccctat aatatgagag\n",
      "    20821 ttatacattt tggtgctggt tctgataaag gagttgcacc aggtacagct gttttaagac\n",
      "    20881 agtggttgcc tacgggtacg ctgcttgtcg attcagatct taatgacttt gtctctgatg\n",
      "    20941 cagattcaac tttgattggt gattgtgcaa ctgtacatac agctaataaa tgggatctca\n",
      "    21001 ttattagtga tatgtacgac cctaagacta aaaatgttac aaaagaaaat gactctaaag\n",
      "    21061 agggtttttt cacttacatt tgtgggttta tacaacaaaa gctagctctt ggaggttccg\n",
      "    21121 tggctataaa gataacagaa cattcttgga atgctgatct ttataagctc atgggacact\n",
      "    21181 tcgcatggtg gacagccttt gttactaatg tgaatgcgtc atcatctgaa gcatttttan\n",
      "    21241 ttggatgtaa ttatcttggc aaaccntgcg nacaaataga tggttatgtc atgcatgcaa\n",
      "    21301 attacatatt ttggaggaat acaaatccaa ttcagttgtc ttcctattct ttatttgaca\n",
      "    21361 tgagtaaatt tccccttaaa ttaaggggta ctgctgttat gtctttaaaa gaaggtcaaa\n",
      "    21421 tcaatgatat gattttatct cttcttagta aaggtagact tataattaga gaaaacaaca\n",
      "    21481 gagttgttat ttctagtgat gttcttgtta acaactaaac gaacaatgtt tgtttttctt\n",
      "    21541 gttttattgc cactagtctc tagtcagtgt gttaatctta caaccagaac tcaattaccc\n",
      "    21601 cctgcataca ctaattcttt cacacgtggt gtttattacc ctgacaaagt tttcagatcc\n",
      "    21661 tcagttttac attcaactca ggacttgttc ttacctttct tttccaatgt tacttggttc\n",
      "    21721 catgctatac atgtctctgg gaccaatggt actaagaggt ttgataaccc tgtcctacca\n",
      "    21781 tttaatgatg gtgtttattt tgcttccact gagaagtcta acataataag aggctggatt\n",
      "    21841 tttggtacta ctttagattc gaagacccag tccctactta ttgttaataa cgctactaat\n",
      "    21901 gttgttatta aagtctgtga atttcaattt tgtaatgatc catttttggg tgtttattac\n",
      "    21961 cacaaaaaca acaaaagttg gatggaaagt gagttcagag tttattctag tgcgaataat\n",
      "    22021 tgcacttttg aatatgtctc tcagcctttt cttatggacc ttgaaggaaa acagggtaat\n",
      "    22081 ttcaaaaatc ttagggaatt tgtgtttaag aatattgatg gttattttaa aatatattct\n",
      "    22141 aagcacacgc ctattaattt agtgcgtgat ctccctcagg gtttttcggc tttagaacca\n",
      "    22201 ttggtagatt tgccaatagg tattaacatc actaggtttc aaactttact tgctttacat\n",
      "    22261 agaagttatt tgactcctgg tgattcttct tcaggttgga cagctggtgc tgcagcttat\n",
      "    22321 tatgtgggtt atcttcaacc taggactttt ctattaaaat ataatgaaaa tggaaccatt\n",
      "    22381 acagatgctg tagactgtgc acttgaccct ctctcagaaa caaagtgtac gttgaaatcc\n",
      "    22441 ttcactgtag aaaaaggaat ctatcaaact tctaacttta gagtccaacc aacagaatct\n",
      "    22501 attgttagat ttcctaatat tacaaacttg tgcccttttg gtgaagtttt taacgccacc\n",
      "    22561 agatttgcat ctgtttatgc ttggaacagg aagagaatca gcaactgtgt tgctgattat\n",
      "    22621 tctgtcctat ataattccgc atcattttcc acttttaagt gttatggagt gtctcctact\n",
      "    22681 aaattaaatg atctctgctt tactaatgtc tatgcagatt catttgtaat tagaggtgat\n",
      "    22741 gaagtcagac aaatcgctcc agggcaaact ggaaagattg ctgattataa ttataaatta\n",
      "    22801 ccagatgatt ttacaggctg cgttatagct tggaattcta acaatcttga ttctaaggtt\n",
      "    22861 ggtggtaatt ataattacct gtatagattg tttaggaagt ctaatctcaa accttttgag\n",
      "    22921 agagatattt caactgaaat ctatcaggcc ggtagcacac cttgtaatgg tgttgaaggt\n",
      "    22981 tttaattgtt actttccttt acaatcatat ggtttccaac ccactaatgg tgttggttac\n",
      "    23041 caaccataca gagtagtagt actttctttt gaacttctac atgcaccagc aactgtttgt\n",
      "    23101 ggacctaaaa agtctactaa tttggttaaa aacaaatgtg tcaatttcaa cttcaatggt\n",
      "    23161 ttaacaggca caggtgttct tactgagtct aacaaaaagt ttctgccttt ccaacaattt\n",
      "    23221 ggcagagaca ttgctgacac tactgatgct gtccgtgatc cacagacact tgagattctt\n",
      "    23281 gacattacac catgttcttt tggtggtgtc agtgttataa caccaggaac aaatacttct\n",
      "    23341 aaccaggttg ctgttcttta tcagggtgtt aactgcacag aagtccctgt tgctattcat\n",
      "    23401 gcagatcaac ttactcctac ttggcgtgtt tattctacag gttctaatgt ttttcaaaca\n",
      "    23461 cgtgcaggct gtttaatagg ggctgaacat gtcaacaact catatgagtg tgacataccc\n",
      "    23521 attggtgcag gtatatgcgc tagttatcag actcagacta attctcctcg gcgggcacgt\n",
      "    23581 agtgtagcta gtcaatccat cattgcctac actatgtcac ttggtgcaga aaattcagtt\n",
      "    23641 gcttactcta ataactctat tgccataccc acaaatttta ctattagtgt taccacagaa\n",
      "    23701 attctaccag tgtctatgac caagacatca gtagattgta caatgtacat ttgtggtgat\n",
      "    23761 tcaactgaat gcagcaatct tttgttgcaa tatggcagtt tttgtacaca attaaaccgt\n",
      "    23821 gctttaactg gaatagctgt tgaacaagac aaaaacaccc aagaagtttt tgcacaagtc\n",
      "    23881 aaacaaattt acaaaacacc accaattaaa gattttggtg gttttaattt ttcacaaata\n",
      "    23941 ttaccagatc catcaaaacc aagcaagagg tcatttattg aagatctact tttcaacaaa\n",
      "    24001 gtgacacttg cagatgctgg cttcatcaaa caatatggtg attgccttgg tgatattgct\n",
      "    24061 gctagagacc tcatttgtgc acaaaagttt aacggcctta ctgttttgcc acctttgctc\n",
      "    24121 acagatgaaa tgattgctca atacacttct gcactgttag cgggtacaat cacttctggt\n",
      "    24181 tggacctttg gtgcaggtgc tgcattacaa ataccatttg ctatgcaaat ggcttatagg\n",
      "    24241 tttaatggta ttggagttac acagaatgtt ctctatgaga accaaaaatt gattgccaac\n",
      "    24301 caatttaata gtgctattgg caaaattcaa gactcacttt cttccacagc aagtgcactt\n",
      "    24361 ggaaaacttc aagatgtggt caaccaaaat gcacaagctt taaacacgct tgttaaacaa\n",
      "    24421 cttagctcca attttggtgc aatttcaagt gttttaaatg atatcctttc acgtcttgac\n",
      "    24481 aaagttgagg ctgaagtgca aattgatagg ttgatcacag gcagacttca aagtttgcag\n",
      "    24541 acatatgtga ctcaacaatt aattagagct gcagaaatca gagcttctgc taatcttgct\n",
      "    24601 gctactaaaa tgtcagagtg tgtacttgga caatcaaaaa gagttgattt ttgtggaaag\n",
      "    24661 ggctatcatc ttatgtcctt ccctcagtca gcacctcatg gtgtagtctt cttgcatgtg\n",
      "    24721 acttatgtcc ctgcacaaga aaagaacttc acaactgctc ctgccatttg tcatgatgga\n",
      "    24781 aaagcacact ttcctcgtga aggtgtcttt gtttcaaatg gcacacactg gtttgtaaca\n",
      "    24841 caaaggaatt tttatgaacc acaaatcatt actacagaca acacatttgt gtctggtaac\n",
      "    24901 tgtgatgttg taataggaat tgtcaacaac acagtttatg atcctttgca acctgaatta\n",
      "    24961 gactcattca aggaggagtt agataaatat tttaagaatc atacatcacc agatgttgat\n",
      "    25021 ttaggtgaca tctctggcat taatgcttca gttgtaaaca ttcaaaaaga aattgaccgc\n",
      "    25081 ctcaatgagg ttgccaagaa tttaaatgaa tctctcatcg atctccaaga acttggaaag\n",
      "    25141 tatgagcagt atataaaatg gccatggtac atttggctag gttttatagc tggcttgatt\n",
      "    25201 gccatagtaa tggtgacaat tatgctttgc tgtatgacca gttgctgtag ttgtctcaag\n",
      "    25261 ggctgttgtt cttgtggatc ctgctgcaaa tttgatgaag acgactctga gccagtgctc\n",
      "    25321 aaaggagtca aattacatta cacataaacg aacttatgga tttgtttatg agaatcttca\n",
      "    25381 caattggaac tgtaactttg aagcaaggtg aaatcaagga tgctactcct tcagattttg\n",
      "    25441 ttcgcgctac tgcaacgata ccgatacaag cctcactccc tttcggatgg cttattgttg\n",
      "    25501 gcgttgcact tcttgctgtt tttcatagcg cttccaaaat cataaccctc aaaaagagat\n",
      "    25561 ggcaactagc actctccaag ggtgttcact ttgtttgcaa cttgctgttg ttgtttgtaa\n",
      "    25621 cagtttactc acaccttttg ctcgttgctg ctggccttga agcccctttt ctctatcttt\n",
      "    25681 atgctttagt ctacttcttg cagagtataa actttgtaag aataataatg aggctttggc\n",
      "    25741 tttgctggaa atgccgttcc aaaaacccat tactttatga tgccaactat tttctttgct\n",
      "    25801 ggcatactaa ttgttacgac tattgtatac cttacaatag tgtaacttct tcaattgtca\n",
      "    25861 ttacttcagt tgatggcaca acaagtccta tttctgaaca tgactaccag attggtggtt\n",
      "    25921 atactgaaaa atgggaatct ggagtaaaag actgtgttgt attacacagt tacttcactt\n",
      "    25981 cagactatta ccagctgtac tcaactcaat tgagtacaga cactggtgtt gaacatgtta\n",
      "    26041 ccttcttcat ctacaataaa attgttgatg agcctgaaga acatgtccaa attcacacaa\n",
      "    26101 tcgacggttc atccggagtt gttaatccag taatggaacc aatttatgat gaaccgacga\n",
      "    26161 cgactactag cgtgcctttg taagcacaag ctgatgagta cgaacttatg tactcattcg\n",
      "    26221 tttcggaaga gacaggtacg ttaatagtta atagcgtact tctttttctt gctttcgtgg\n",
      "    26281 tattcttgct agttacacta gccatcctta ctgcgcttcg attgtgtgcg tactgctgca\n",
      "    26341 atattgttaa cgtgagtctt gtaaaacctt ctttttacgt ttactctcgt gttaaaaatc\n",
      "    26401 tgaattcttc tagagttcct gatcttctgg tctaaacgaa ctaaatatta tattagtttt\n",
      "    26461 tctgtttgga actttaattt tagccatggc agattccaac ggtactatta ccgttgaaga\n",
      "    26521 gcttaaaaag ctccttgaac aatggaacct agtaataggt ttcctattcc ttacatggat\n",
      "    26581 ttgtcttcta caatttgcct atgccaacag gaataggttt ttgtatataa ttaagttaat\n",
      "    26641 tttcctctgg ctgttatggc cagtaacttt agcttgtttt gtgcttgctg ctgtttacag\n",
      "    26701 aataaattgg atcaccggtg gaattgctat cgcaatggct tgtcttgtag gcttgatgtg\n",
      "    26761 gctcagctac ttcattgctt cttttagact gtttgcgcgt acgcgttcca tgtggtcatt\n",
      "    26821 caatccagaa actaacattc ttctcaacgt gccactccat ggcactattc tgaccagacc\n",
      "    26881 gcttctagaa agtgaactcg taatcggagc tgtgatcctt cgtggacatc ttcgtattgc\n",
      "    26941 tggacaccat ctaggacgct gtgacatcaa ggacctgcct aaagaaatca ctgttgctac\n",
      "    27001 atcacgaacg ctttcttatt acaaattggg agcttcgcag cgtgtagcag gtgactcagg\n",
      "    27061 ttttgctgca tacagtcgct acaggattgg caactataaa ttaaacacag accattccag\n",
      "    27121 tagcagtgac aatattgctt tgcttgtaca gtaagtgaca acagatgttt catctcgttg\n",
      "    27181 actttcaggt tactatagca gagatattac taattattat gaggactttt aaagtttcca\n",
      "    27241 tttggaatct tgattacatc ataaacctca taattaaaaa tttatctaag tcactaactg\n",
      "    27301 agaataaata ttctcaatta gatgaagagc aaccaatgga gattgattaa acgaacatga\n",
      "    27361 aaattattct tttcttggca ctgataacac tcgctacttg tgagctttat cactaccaag\n",
      "    27421 agtgtgttag aggtacaaca gtacttttaa aagaaccttg ctcttctgga acatacgagg\n",
      "    27481 gcaattcacc atttcatcct ctagctgata acanatttgc actgacttgc tttagcactc\n",
      "    27541 aatttgcttt tncttntccn nnnngcgtaa aacacgtcta tcagttacgt gcnnnnnnag\n",
      "    27601 tttcacctaa actgttcatc agacaagagg aagttnaann actttactct ccantntttc\n",
      "    27661 ttattgttgc ngcaanagtg nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn\n",
      "    27721 nnnnnnnnnn nnnnnnnnnn nnnnnnnnnn nnnnttttta gcctttctgc tattccttgt\n",
      "    27781 tttaattatg cttattatct tttggttctc acttgaactg caagatcata atgaaacttg\n",
      "    27841 tcacgcctaa acgaacatga aatttcttgt tttcttagga atcatcacaa ctgtagctgc\n",
      "    27901 atttcaccaa gaatgtagtt tacagttatg tactcaacat caaccatatg tagttgatga\n",
      "    27961 cccgtgtcct attcacttct attctaaatg gtatattaga gtaggagcta gaaaatcagc\n",
      "    28021 acctttaatt gaattgtgcg tggatgaggc tggttctaaa tcacccattc agtacatcga\n",
      "    28081 tatcggtaat tatacagttt cctgtttacc ttttacaatt aattgccagg aacctaaatt\n",
      "    28141 gggtagtctt gtagtgcgtt gttcgttcta tgaagacttt ttagagtatc atgacgttcg\n",
      "    28201 tgttgtttta gatttcatct aaacgaacaa actaaaatgt ctgataatgg accccaaaat\n",
      "    28261 cagcgaaatg caccccgcat tacgtttggt ggaccctcag attcaactgg cagtaaccag\n",
      "    28321 aatggagaac gcagtggggc gcgatcaaaa caacgtcggc cccaaggttt acccaataat\n",
      "    28381 actgcgtctt ggttcaccgc tctcactcaa catggcaagg aagaccttaa attctctcga\n",
      "    28441 ggacaaggcg ttccaattaa caccaatagc agtccagatg accaaattgg ctactaccga\n",
      "    28501 agagctacca gacgaattcg tggtggtgac ggtaaaatga aagatctcag tccaagatgg\n",
      "    28561 tatttctact acctaggaac tgggccagaa gctggacttc cctatggtgc taacaaagac\n",
      "    28621 ggcatcatat gggttgcaac tgagggagcc ttgaatacac caaaagatca cattggcacc\n",
      "    28681 cgcaatcctg ctaacaatgc tgcaatcgtg ctacaacttc ctcaaggaac aacattgcca\n",
      "    28741 aaaggcttct acgcagaagg gagcagaggc ggcagtcaag cctcttctcg ttcctcatca\n",
      "    28801 cgtagtcgca acagttcaag aaattcaact ctaggcagca gtaggggaac ttctcctgct\n",
      "    28861 agaatggctg gcaatggcgg tgatgctgct cttgctttgc tgctgcttga cagattgaac\n",
      "    28921 cagcttgaga gcaaaatgtc tggtaaaggc caacaacaac aaggccaaac tgtcactaag\n",
      "    28981 aaatctgctg ctgaggcttc taagaagcct cggcaaaaac gtactgccac taaagcatac\n",
      "    29041 aatgtaacac aagctttcgg cagacgtggt ccagaacaaa cccaaggaaa ttttggggac\n",
      "    29101 caggaactaa tcagacaagg aactgattac aaacattggc cgcaaattgc acaatttgcc\n",
      "    29161 cccagcgctt cagcgttctt cggaatgtcg cgcattggca tggaagtcac accttcggga\n",
      "    29221 acgtggttga cctacacagg tgccatcaaa ttggatgaca aagatccaaa tttcaaagat\n",
      "    29281 caagtcattt tgctgaataa gcatattgac gcatacaaaa cattcccacc aacagagcct\n",
      "    29341 aaaaaggaca aaaagaagaa ggctgatgaa actcaagcct taccgcagag acagaagaaa\n",
      "    29401 cagcaaactg tgactcttct tcctgctgca gatttggatg atttctccaa acaattgcaa\n",
      "    29461 caatccatga gcagtgctga ctcaactcag gcctaaactc atgcagacca cacaaggcag\n",
      "    29521 atgggctata taaacgtttt cgcttttccg tttacgatat atagtctact cttgtgcaga\n",
      "    29581 atgaattctc gtaactacat agcacaagta gatgtagtta actttaatct cacatagcaa\n",
      "    29641 tctttaatca gtgtgtaaca ttagggagga cttgaaagag ccaccacatt ttcaccgagg\n",
      "    29701 ccacgcggag tacgatcgag tgtacagtga acaatgctag ggagagctgc ctatatggaa\n",
      "    29761 gagccctaat gtgtaaaatt aattttagta gtgctatccc catgtgattt taatagcttc\n",
      "    29821 ttaggagaat ga\n",
      "//\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "handle =Entrez.efetch(db=\"nucleotide\",id=\"1993774296\",rettype='gb' ,retmode=\"text\")\n",
    "print(handle.read())\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "handle = Entrez.esearch(db='nucleotide', term='accD[Gene Name] AND \"E. coli\"[Organism]', retmax=\"20\")\n",
    "result_list = Entrez.read(handle)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['2727925337', '2727925307', '2727924808', '2727924260', '2727923901', '2727923899', '2727682446', '2727614230', '2727614229', '2727614228', '2727613749', '2727613748', '2727613747', '2727613727', '2727581472', '2727577636', '2727577630', '2727577629', '2727577627', '2727577625']\n",
      "\n",
      "\n",
      "267924\n"
     ]
    }
   ],
   "source": [
    "id_list = result_list['IdList']\n",
    "count = result_list['Count']\n",
    "\n",
    "print(id_list)\n",
    "print(\"\\n\")\n",
    "print(count)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "handle.close() \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "------------------------------------------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 3. PDB"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import Modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.PDB import PDBParser,PDBList"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on class PDBList in module Bio.PDB.PDBList:\n",
      "\n",
      "class PDBList(builtins.object)\n",
      " |  Quick access to the structure lists on the PDB or its mirrors.\n",
      " |  \n",
      " |  This class provides quick access to the structure lists on the\n",
      " |  PDB server or its mirrors. The structure lists contain\n",
      " |  four-letter PDB codes, indicating that structures are\n",
      " |  new, have been modified or are obsolete. The lists are released\n",
      " |  on a weekly basis.\n",
      " |  \n",
      " |  It also provides a function to retrieve PDB files from the server.\n",
      " |  To use it properly, prepare a directory /pdb or the like,\n",
      " |  where PDB files are stored.\n",
      " |  \n",
      " |  All available file formats (PDB, PDBx/mmCif, PDBML, mmtf) are supported.\n",
      " |  Please note that large structures (containing >62 chains\n",
      " |  and/or 99999 ATOM lines) are no longer stored as a single PDB file\n",
      " |  and by default (when PDB format selected) are not downloaded.\n",
      " |  \n",
      " |  Large structures can be downloaded in other formats, including PDBx/mmCif\n",
      " |  or as a .tar file (a collection of PDB-like formatted files for a given\n",
      " |  structure).\n",
      " |  \n",
      " |  If you want to use this module from inside a proxy, add\n",
      " |  the proxy variable to your environment, e.g. in Unix:\n",
      " |  export HTTP_PROXY='http://realproxy.charite.de:888'\n",
      " |  (This can also be added to ~/.bashrc)\n",
      " |  \n",
      " |  Methods defined here:\n",
      " |  \n",
      " |  __init__(self, server='ftp://ftp.wwpdb.org', pdb=None, obsolete_pdb=None, verbose=True)\n",
      " |      Initialize the class with the default server or a custom one.\n",
      " |      \n",
      " |      Argument pdb is the local path to use, defaulting to the current\n",
      " |      directory at the moment of initialisation.\n",
      " |  \n",
      " |  download_entire_pdb(self, listfile=None, file_format=None)\n",
      " |      Retrieve all PDB entries not present in the local PDB copy.\n",
      " |      \n",
      " |      :param listfile: filename to which all PDB codes will be written (optional)\n",
      " |      \n",
      " |      :param file_format:\n",
      " |          File format. Available options:\n",
      " |      \n",
      " |          * \"mmCif\" (default, PDBx/mmCif file),\n",
      " |          * \"pdb\" (format PDB),\n",
      " |          * \"xml\" (PMDML/XML format),\n",
      " |          * \"mmtf\" (highly compressed),\n",
      " |          * \"bundle\" (PDB formatted archive for large structure}\n",
      " |      \n",
      " |      NOTE. The default download format has changed from PDB to PDBx/mmCif\n",
      " |  \n",
      " |  download_obsolete_entries(self, listfile=None, file_format=None)\n",
      " |      Retrieve all obsolete PDB entries not present in local obsolete PDB copy.\n",
      " |      \n",
      " |      :param listfile: filename to which all PDB codes will be written (optional)\n",
      " |      \n",
      " |      :param file_format: file format. Available options:\n",
      " |          \"mmCif\" (default, PDBx/mmCif file),\n",
      " |          \"pdb\" (format PDB),\n",
      " |          \"xml\" (PMDML/XML format),\n",
      " |      \n",
      " |      NOTE. The default download format has changed from PDB to PDBx/mmCif\n",
      " |  \n",
      " |  download_pdb_files(self, pdb_codes, obsolete=False, pdir=None, file_format=None, overwrite=False)\n",
      " |      Fetch set of PDB structure files from the PDB server and stores them locally.\n",
      " |      \n",
      " |      The PDB structure's file name is returned as a single string.\n",
      " |      If obsolete ``==`` True, the files will be saved in a special file tree.\n",
      " |      \n",
      " |      :param pdb_codes: a list of 4-symbols structure Ids from PDB\n",
      " |      :type pdb_codes: list of strings\n",
      " |      \n",
      " |      :param file_format:\n",
      " |          File format. Available options:\n",
      " |      \n",
      " |          * \"mmCif\" (default, PDBx/mmCif file),\n",
      " |          * \"pdb\" (format PDB),\n",
      " |          * \"xml\" (PMDML/XML format),\n",
      " |          * \"mmtf\" (highly compressed),\n",
      " |          * \"bundle\" (PDB formatted archive for large structure}\n",
      " |      \n",
      " |      :param overwrite: if set to True, existing structure files will be overwritten. Default: False\n",
      " |      :type overwrite: bool\n",
      " |      \n",
      " |      :param obsolete:\n",
      " |          Has a meaning only for obsolete structures.\n",
      " |          If True, download the obsolete structure\n",
      " |          to 'obsolete' folder, otherwise download won't be performed.\n",
      " |          This option doesn't work for mmtf format as obsoleted structures are not availbe as mmtf.\n",
      " |          (default: False)\n",
      " |      \n",
      " |      :type obsolete: bool\n",
      " |      \n",
      " |      :param pdir: put the file in this directory (default: create a PDB-style directory tree)\n",
      " |      :type pdir: string\n",
      " |      \n",
      " |      :return: filenames\n",
      " |      :rtype: string\n",
      " |  \n",
      " |  get_all_entries(self)\n",
      " |      Retrieve the big file containing all the PDB entries and some annotation.\n",
      " |      \n",
      " |      Returns a list of PDB codes in the index file.\n",
      " |  \n",
      " |  get_all_obsolete(self)\n",
      " |      Return a list of all obsolete entries ever in the PDB.\n",
      " |      \n",
      " |      Returns a list of all obsolete pdb codes that have ever been\n",
      " |      in the PDB.\n",
      " |      \n",
      " |      Gets and parses the file from the PDB server in the format\n",
      " |      (the first pdb_code column is the one used). The file looks\n",
      " |      like this::\n",
      " |      \n",
      " |           LIST OF OBSOLETE COORDINATE ENTRIES AND SUCCESSORS\n",
      " |          OBSLTE    31-JUL-94 116L     216L\n",
      " |          ...\n",
      " |          OBSLTE    29-JAN-96 1HFT     2HFT\n",
      " |          OBSLTE    21-SEP-06 1HFV     2J5X\n",
      " |          OBSLTE    21-NOV-03 1HG6\n",
      " |          OBSLTE    18-JUL-84 1HHB     2HHB 3HHB\n",
      " |          OBSLTE    08-NOV-96 1HID     2HID\n",
      " |          OBSLTE    01-APR-97 1HIU     2HIU\n",
      " |          OBSLTE    14-JAN-04 1HKE     1UUZ\n",
      " |          ...\n",
      " |  \n",
      " |  get_recent_changes(self)\n",
      " |      Return three lists of the newest weekly files (added,mod,obsolete).\n",
      " |      \n",
      " |      Reads the directories with changed entries from the PDB server and\n",
      " |      returns a tuple of three URL's to the files of new, modified and\n",
      " |      obsolete entries from the most recent list. The directory with the\n",
      " |      largest numerical name is used.\n",
      " |      Returns None if something goes wrong.\n",
      " |      \n",
      " |      Contents of the data/status dir (20031013 would be used);:\n",
      " |      \n",
      " |          drwxrwxr-x   2 1002     sysadmin     512 Oct  6 18:28 20031006\n",
      " |          drwxrwxr-x   2 1002     sysadmin     512 Oct 14 02:14 20031013\n",
      " |          -rw-r--r--   1 1002     sysadmin    1327 Mar 12  2001 README\n",
      " |  \n",
      " |  get_seqres_file(self, savefile='pdb_seqres.txt')\n",
      " |      Retrieve and save a (big) file containing all the sequences of PDB entries.\n",
      " |  \n",
      " |  retrieve_pdb_file(self, pdb_code, obsolete=False, pdir=None, file_format=None, overwrite=False)\n",
      " |      Fetch PDB structure file from PDB server, and store it locally.\n",
      " |      \n",
      " |      The PDB structure's file name is returned as a single string.\n",
      " |      If obsolete ``==`` True, the file will be saved in a special file tree.\n",
      " |      \n",
      " |      NOTE. The default download format has changed from PDB to PDBx/mmCif\n",
      " |      \n",
      " |      :param pdb_code: 4-symbols structure Id from PDB (e.g. 3J92).\n",
      " |      :type pdb_code: string\n",
      " |      \n",
      " |      :param file_format:\n",
      " |          File format. Available options:\n",
      " |      \n",
      " |          * \"mmCif\" (default, PDBx/mmCif file),\n",
      " |          * \"pdb\" (format PDB),\n",
      " |          * \"xml\" (PDBML/XML format),\n",
      " |          * \"mmtf\" (highly compressed),\n",
      " |          * \"bundle\" (PDB formatted archive for large structure}\n",
      " |      \n",
      " |      :type file_format: string\n",
      " |      \n",
      " |      :param overwrite: if set to True, existing structure files will be overwritten. Default: False\n",
      " |      :type overwrite: bool\n",
      " |      \n",
      " |      :param obsolete:\n",
      " |          Has a meaning only for obsolete structures. If True, download the obsolete structure\n",
      " |          to 'obsolete' folder, otherwise download won't be performed.\n",
      " |          This option doesn't work for mmtf format as obsoleted structures aren't stored in mmtf.\n",
      " |          Also doesn't have meaning when parameter pdir is specified.\n",
      " |          Note: make sure that you are about to download the really obsolete structure.\n",
      " |          Trying to download non-obsolete structure into obsolete folder will not work\n",
      " |          and you face the \"structure doesn't exists\" error.\n",
      " |          Default: False\n",
      " |      \n",
      " |      :type obsolete: bool\n",
      " |      \n",
      " |      :param pdir: put the file in this directory (default: create a PDB-style directory tree)\n",
      " |      :type pdir: string\n",
      " |      \n",
      " |      :return: filename\n",
      " |      :rtype: string\n",
      " |  \n",
      " |  update_pdb(self, file_format=None)\n",
      " |      Update your local copy of the PDB files.\n",
      " |      \n",
      " |      I guess this is the 'most wanted' function from this module.\n",
      " |      It gets the weekly lists of new and modified pdb entries and\n",
      " |      automatically downloads the according PDB files.\n",
      " |      You can call this module as a weekly cron job.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Static methods defined here:\n",
      " |  \n",
      " |  get_status_list(url)\n",
      " |      Retrieve a list of pdb codes in the weekly pdb status file from given URL.\n",
      " |      \n",
      " |      Used by get_recent_changes. Typical contents of the list files parsed\n",
      " |      by this method is now very simply - one PDB name per line.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data descriptors defined here:\n",
      " |  \n",
      " |  __dict__\n",
      " |      dictionary for instance variables (if defined)\n",
      " |  \n",
      " |  __weakref__\n",
      " |      list of weak references to the object (if defined)\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data and other attributes defined here:\n",
      " |  \n",
      " |  PDB_REF = '\\n    The Protein Data Bank: a computer-based arc... pp. 53...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(PDBList)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Structure exists: 'dir/pdb7byr.ent' \n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'dir/pdb7byr.ent'"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pdbl=PDBList()\n",
    "pdbl.retrieve_pdb_file(\"7BYR\",file_format=\"pdb\",pdir=\"dir\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/rhyme/.local/lib/python3.6/site-packages/Bio/PDB/StructureBuilder.py:92: PDBConstructionWarning: WARNING: Chain A is discontinuous at line 26237.\n",
      "  PDBConstructionWarning,\n",
      "/home/rhyme/.local/lib/python3.6/site-packages/Bio/PDB/StructureBuilder.py:92: PDBConstructionWarning: WARNING: Chain B is discontinuous at line 26405.\n",
      "  PDBConstructionWarning,\n",
      "/home/rhyme/.local/lib/python3.6/site-packages/Bio/PDB/StructureBuilder.py:92: PDBConstructionWarning: WARNING: Chain C is discontinuous at line 26545.\n",
      "  PDBConstructionWarning,\n"
     ]
    }
   ],
   "source": [
    "parser =PDBParser()\n",
    "structure =parser.get_structure(\"7BYR\",\"dir/pdb7byr.ent\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chaibid :A\n",
      "chaibid :B\n",
      "chaibid :C\n",
      "chaibid :H\n",
      "chaibid :L\n",
      "chaibid :D\n",
      "chaibid :E\n",
      "chaibid :F\n",
      "chaibid :G\n",
      "chaibid :I\n",
      "chaibid :J\n"
     ]
    }
   ],
   "source": [
    "for chain in structure[0]:\n",
    "    print(f\"chaibid :{chain.id}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3.84"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "resolution = structure.header[\"resolution\"]\n",
    "resolution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'sars-cov-2, antigen, rbd, neutralizing antibody, viral protein'"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "keywords =structure.header[\"keywords\"]\n",
    "keywords"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "------------------------------------------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 4. EXPASY"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4.1. PROSITE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import Modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import ExPASy\n",
    "from Bio.ExPASy import Prosite"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on module Bio.ExPASy.Prosite in Bio.ExPASy:\n",
      "\n",
      "NAME\n",
      "    Bio.ExPASy.Prosite - Parser for the prosite dat file from Prosite at ExPASy.\n",
      "\n",
      "DESCRIPTION\n",
      "    See https://www.expasy.org/prosite/\n",
      "    \n",
      "    Tested with:\n",
      "     - Release 20.43, 10-Feb-2009\n",
      "     - Release 2017_03 of 15-Mar-2017.\n",
      "    \n",
      "    Functions:\n",
      "     - read                  Reads a Prosite file containing one Prosite record\n",
      "     - parse                 Iterates over records in a Prosite file.\n",
      "    \n",
      "    Classes:\n",
      "     - Record                Holds Prosite data.\n",
      "\n",
      "CLASSES\n",
      "    builtins.object\n",
      "        Record\n",
      "    \n",
      "    class Record(builtins.object)\n",
      "     |  Holds information from a Prosite record.\n",
      "     |  \n",
      "     |  Main attributes:\n",
      "     |   - name           ID of the record.  e.g. ADH_ZINC\n",
      "     |   - type           Type of entry.  e.g. PATTERN, MATRIX, or RULE\n",
      "     |   - accession      e.g. PS00387\n",
      "     |   - created        Date the entry was created.  (MMM-YYYY for releases\n",
      "     |     before January 2017, DD-MMM-YYYY since January 2017)\n",
      "     |   - data_update    Date the 'primary' data was last updated.\n",
      "     |   - info_update    Date data other than 'primary' data was last updated.\n",
      "     |   - pdoc           ID of the PROSITE DOCumentation.\n",
      "     |   - description    Free-format description.\n",
      "     |   - pattern        The PROSITE pattern.  See docs.\n",
      "     |   - matrix         List of strings that describes a matrix entry.\n",
      "     |   - rules          List of rule definitions (from RU lines).  (strings)\n",
      "     |   - prorules       List of prorules (from PR lines). (strings)\n",
      "     |  \n",
      "     |  NUMERICAL RESULTS:\n",
      "     |   - nr_sp_release  SwissProt release.\n",
      "     |   - nr_sp_seqs     Number of seqs in that release of Swiss-Prot. (int)\n",
      "     |   - nr_total       Number of hits in Swiss-Prot.  tuple of (hits, seqs)\n",
      "     |   - nr_positive    True positives.  tuple of (hits, seqs)\n",
      "     |   - nr_unknown     Could be positives.  tuple of (hits, seqs)\n",
      "     |   - nr_false_pos   False positives.  tuple of (hits, seqs)\n",
      "     |   - nr_false_neg   False negatives.  (int)\n",
      "     |   - nr_partial     False negatives, because they are fragments. (int)\n",
      "     |  \n",
      "     |  COMMENTS:\n",
      "     |   - cc_taxo_range  Taxonomic range.  See docs for format\n",
      "     |   - cc_max_repeat  Maximum number of repetitions in a protein\n",
      "     |   - cc_site        Interesting site.  list of tuples (pattern pos, desc.)\n",
      "     |   - cc_skip_flag   Can this entry be ignored?\n",
      "     |   - cc_matrix_type\n",
      "     |   - cc_scaling_db\n",
      "     |   - cc_author\n",
      "     |   - cc_ft_key\n",
      "     |   - cc_ft_desc\n",
      "     |   - cc_version     version number (introduced in release 19.0)\n",
      "     |  \n",
      "     |  The following are all lists if tuples (swiss-prot accession, swiss-prot name).\n",
      "     |  \n",
      "     |  DATA BANK REFERENCES:\n",
      "     |   - dr_positive\n",
      "     |   - dr_false_neg\n",
      "     |   - dr_false_pos\n",
      "     |   - dr_potential   Potential hits, but fingerprint region not yet available.\n",
      "     |   - dr_unknown     Could possibly belong\n",
      "     |   - pdb_structs    List of PDB entries.\n",
      "     |  \n",
      "     |  Methods defined here:\n",
      "     |  \n",
      "     |  __init__(self)\n",
      "     |      Initialize the class.\n",
      "     |  \n",
      "     |  ----------------------------------------------------------------------\n",
      "     |  Data descriptors defined here:\n",
      "     |  \n",
      "     |  __dict__\n",
      "     |      dictionary for instance variables (if defined)\n",
      "     |  \n",
      "     |  __weakref__\n",
      "     |      list of weak references to the object (if defined)\n",
      "\n",
      "FUNCTIONS\n",
      "    parse(handle)\n",
      "        Parse Prosite records.\n",
      "        \n",
      "        This function is for parsing Prosite files containing multiple\n",
      "        records.\n",
      "        \n",
      "        Arguments:\n",
      "         - handle   - handle to the file.\n",
      "    \n",
      "    read(handle)\n",
      "        Read one Prosite record.\n",
      "        \n",
      "        This function is for parsing Prosite files containing\n",
      "        exactly one record.\n",
      "        \n",
      "        Arguments:\n",
      "         - handle   - handle to the file.\n",
      "\n",
      "FILE\n",
      "    /home/rhyme/.local/lib/python3.6/site-packages/Bio/ExPASy/Prosite.py\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(Prosite)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "handle = ExPASy.get_prosite_raw(\"PS51442\")\n",
    "record = Prosite.read(handle)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Coronavirus main protease (M-pro) domain profile.\n"
     ]
    }
   ],
   "source": [
    "print(record.description)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['1LVO', '1P9S', '1P9U', '1Q2W', '1UJ1', '1UK2', '1UK3', '1UK4', '1WOF', '1Z1I']\n"
     ]
    }
   ],
   "source": [
    "print(record.pdb_structs[:10])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Unknown keyword <! found",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-40-56ba8b763590>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mhandle\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mExPASy\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_prosite_raw\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'P500001'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mrecord\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mProsite\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrecord\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpattern\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.6/site-packages/Bio/ExPASy/Prosite.py\u001b[0m in \u001b[0;36mread\u001b[0;34m(handle)\u001b[0m\n\u001b[1;32m     51\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     52\u001b[0m     \"\"\"\n\u001b[0;32m---> 53\u001b[0;31m     \u001b[0mrecord\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m__read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     54\u001b[0m     \u001b[0;31m# We should have reached the end of the record by now\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     55\u001b[0m     \u001b[0mremainder\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhandle\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.6/site-packages/Bio/ExPASy/Prosite.py\u001b[0m in \u001b[0;36m__read\u001b[0;34m(handle)\u001b[0m\n\u001b[1;32m    309\u001b[0m             \u001b[0;32mbreak\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    310\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 311\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Unknown keyword %s found\"\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0mkeyword\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    312\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    313\u001b[0m         \u001b[0;32mreturn\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: Unknown keyword <! found"
     ]
    }
   ],
   "source": [
    "handle = ExPASy.get_prosite_raw('P500001')\n",
    "record = Prosite.read(handle)\n",
    "print(record.pattern)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4.2. ScanProsite"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import Modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.ExPASy import ScanProsite"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "prot_record = SeqIO.read(\"prot_seq.fasta\", format=\"fasta\")\n",
    "len(prot_record.seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "handle = ScanProiste(seq=prot_record.seq, mirror=\"https://prosite.expasy.org/\")\n",
    "result = ScanProiste.read(handle)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "result.n_match"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "result[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "------------------------------------------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 5. KEGG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import Modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.KEGG import REST, Enzyme"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "help(Enzyme)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "request=REST.kegg_get(\"ec:5.4.22\")\n",
    "open(\"ec_5.4.2.2.txt\",\"w\").write(request.read())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "records=Enzyme.parse(open(\"ec_5.4.2.2.txt\"))\n",
    "record =List(records)[]\n",
    "record.classname"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "recorf.pathway"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "record.genes[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "list_genes=[]\n",
    "for x,y in record.genes:\n",
    "    list_genes +=x.split(\"\\n\")\n",
    "    \n",
    "print(list_genes[:10])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "------------------------------------------------------"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
